---
title: 'ATAC-seq-analysis: Reprogramming of fibroblasts'
author:
  - name: Ali T. Abdallah ^[CECAD Research Center - University of Cologne, ali.abdallah@rwth-aachen.de]
fontsize: 15pt
date: "`r Sys.Date()`"
output: 
  html_notebook:
  toc: yes
  highlight: kate
  theme: yeti
  css: "style/theme.css"
---

<style>
.main-container{
    min-width: 1800px;!important
    margin-left: auto;!important
    margin-right: auto;!important
    font-size: 15px;!important
}

#toc_container {
    background: #f9f9f9 none repeat scroll 0 0;
    border: 0px solid #aaa;
    display: table;
    font-size: 100%;
    margin-bottom: 1em;
    padding: 10px;
    width: auto;
}

.toc_title {
    font-weight: 700;
    text-align: left;
}

#toc_container li, #toc_container ol, #toc_container ol li{
    list-style: outside none none !important;
}

h3{
  color: #800000;
  font-weight: bold;
}

h4{
  color: #000080;
  font-weight: bold;
}

h4 a{
  color: #000080;
}

h5{
  color: #008080;
  font-weight: bold
}

h5 a{
  color: #008080;
}

</style>


<div id="toc_container">
<p class="toc_title" id="back"><h4 style="color:#800000"><b>Contents</b></h4></p>
<ol class="toc_list" style="font-size:15px">
  
  <li><a href="#QC"><b>1. Quality Control</b></a>
      <ol class="toc_list">
      <li><a href="#QCfld">1.1 Fragement length distribution </a></li>
      <li><a href="#QCpcash">1.2 PCA analysis and similarity heatmap </a></li>
    </ol>
  </li>
  
  <li><a href="#DAA"><b>2. Differential Accessibility analysis</b></a> 
    <ol class="toc_list">
      <li><a href="#DAAt">2.1 Tables</a></li>
      <li><a href="#DAAvp">2.2 Volcano plots</a></li>
      <li><a href="#DAAhm">2.3 Heatmaps</a></li>
    </ol>
  </li>

  <li><a href="#GSEA"><b>3. GO enrichment analysis </b></a>  </li>
  
  <li><a href="#Integration"><b>4. Omics Integration</b></a></li>
  
  <li><a href="#MotifanalysisFootprinting"><b>5. Motif matching and footprinting </b></a>
      <ol class="toc_list">
      <li><a href="#MotifanalysisFootprintingma">5.1 Motif analysis </a></li>
      <li><a href="#MotifanalysisFootprintingfp">5.2 footprinting </a></li>
      </ol>
  </li>
</ol>
</div>



## {.tabset .tabset-fade}


```{r setup, echo=F}
source("./R/functions.R")
source("./R/NGSTools.R")
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(targets)
library(ggplot2)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
txdb$user_seqlevels <- gsub(pattern="chr", replacement = "", txdb$user_seqlevels)
tar_make()
print("")
```


<h3 id="QC"><a href="#back"> 1. Quality Control </a></h3>

<h4 id="QCfld"> <a href="#back"> 1.1 Fragment length distribution </a></h4>
None

<h4 id="QCpcash"> <a href="#back">  1.2 PCA and similarity heatmap </a></h4>

### PCA and similarity heatmap
```{r fig.width=12, fig.height=5, echo=F}
colData <- tar_read(colData)

pdf(file = "paper/plots/pca.and.simhm.pdf", width = 12, height = 5)
ggplotify::as.ggplot(cowplot::plot_grid(tar_read(PCA_plot), tar_read(SIM_hm), ncol=2)) + #
  ggtitle(paste(unique(colData$condition)[1], " vs. ", unique(colData$condition)[2], sep="")) + 
  theme(plot.title=element_text(face = "bold", hjust=0.5, colour = "#000080"))
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",pca_p, sep="")),
  output_name = pca_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```

### Meta.Data
```{r echo=F}
Table_title <- "Metadata of Lyn.KO experiment"
table_name <- "Lyn.Ko.Meta.Data"
DT::datatable(colData, rownames = T, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

## {.tabset .tabset-fade}
<h3 id="DAA"><a href="#back">2. Differential accessibility analysis </a></h3>

<h4 id="DAAt"><a href="#back">2.1 Tables</a></h4>

### Promoter
```{r echo=F, warning=F}
Table_title <- "Table of differentially accessible genes (by interval - promoter region)"
table_name <- "Diff_accessible_genes_by_interval_promoter"
sign.results <- tar_read(sign.results)
DT::datatable(sign.results[sign.results$reduced.annotation %in% c("Promoter"), ], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))

```

### All
```{r echo=F, warning=F}
Table_title <- "Table of differentially accessible genes (by interval)"
table_name <- "Diff_accessible_genes_by_interval"
DT::datatable(sign.results, rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))




```

```{r echo=F}
generate_motif_dataframe_bed <- function(annotatedPeaks.df, results){
  sr <- results %>% dplyr::filter(Geneid %in% annotatedPeaks.df$Geneid) %>%  dplyr::select(Geneid, Symbol, log2FoldChange, padj) %>% na.omit()
  motif.bed.file <- annotatedPeaks.df %>% 
                    dplyr::filter(Geneid %in% results$Geneid) %>%  
                    dplyr::select(seqnames, start, end, Geneid, Symbol) %>% 
                    na.omit() %>% 
                    merge(sr, by=c("Geneid", "Symbol")) %>%
                    relocate(Geneid, Symbol, .after=end) %>% 
                    arrange(seqnames, start, end) 
  return(motif.bed.file)
}

d1 <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)

target_dir <- "~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedReplicate/indir/motif.analysis/"
roi <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)
bg <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = results)
write.table(roi, file=paste(target_dir,"/","roi.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
write.table(bg, file=paste(target_dir,"/","bg.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
```


```{r echo=F}
if(F){
  generate_motif_dataframe_bed <- function(annotatedPeaks.df, dap.df, seq.colum.name="seqnames", 
                                           start.column.name="start", end.column.name="end",
                                           peak.column.name="Geneid", gene.column.name="Symbol", 
                                           l2fc.column.name="log2FoldChange", padj.column.name="padj"){
    
    sr <- dap.df %>% 
          dplyr::filter(get(peak.column.name) %in% annotatedPeaks.df[,peak.column.name]) %>% 
          dplyr::select(peak.column.name, gene.column.name, l2fc.column.name, padj.column.name) %>%
          na.omit()
    motif.bed.file <- annotatedPeaks.df %>% 
                      dplyr::filter(get(peak.column.name) %in% dap.df[,peak.column.name]) %>%  
                      dplyr::select(seq.colum.name, start.column.name, end.column.name, peak.column.name, gene.column.name) %>% 
                      na.omit() %>% 
                      merge(sr, by=c(peak.column.name, gene.column.name)) %>%
                      relocate(peak.column.name, gene.column.name, .after=end.column.name) %>% 
                      arrange(get(seq.colum.name), get(start.column.name), get(end.column.name) )
    return(motif.bed.file)
  }
  
 d2 <-  generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, dap.df = sign.results)
   
  target_dir <- "~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedReplicate/indir/motif.analysis/"
  roi <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)
  bg <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = results)
  write.table(roi, file=paste(target_dir,"/","roi.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
  write.table(bg, file=paste(target_dir,"/","bg.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
}
```

## {.tabset .tabset-fade}

<h4 id="DAAvp"> <a href="#back">2.2 Volcano Plots </a></h4>

### Combined
```{r fig.height=10, fig.width=21, warning=F, echo=F}
vpi <- tar_read(VPI)
vpt <- tar_read(VPT)
vpg <- tar_read(VPG)
ggplotify::as.ggplot(plot_grid(ggplot()+theme_void(),plot_grid(vpi, vpt, ncol=2), rel_heights = c(1,20), ncol=1)) + ggtitle("Volcano plots at both levels (Peak, Transcript)", subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + theme(plot.title = element_text(face="bold", colour = "#000080", hjust = 0.5, size=18), plot.subtitle = element_text(face="plain", colour = "black", hjust = 0.5, size=12))


```

### Intervall
```{r fig.height=6, fig.width=7, warning=F, echo=F}
pdf(file = "paper/plots/volcano.plot.intervall.pdf", width = 7, height = 6)
vpi+ 
  ggtitle("Volcano plot at peak level", 
          subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + 
  theme(plot.title =
          element_text(face="bold", 
                       colour = "#000080", 
                       hjust = 0.5, size=18), 
        plot.subtitle = 
          element_text(face="plain", 
                       colour = "black", 
                       hjust = 0.5, 
                       size=12))
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",vpi_p, sep="")),
  output_name = vpi_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```

### Transcript
```{r fig.height=6, fig.width=7, warning=F, echo=F}
pdf(file = "paper/plots/volcano.plot.transcript.pdf", width = 7, height = 6)
vpt+ 
  ggtitle("Volcano plot at transcript level", 
          subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + 
  theme(plot.title =
          element_text(face="bold", 
                       colour = "#000080", 
                       hjust = 0.5, size=18), 
        plot.subtitle = 
          element_text(face="plain", 
                       colour = "black", 
                       hjust = 0.5, 
                       size=12))
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",vpt_p, sep="")),
  output_name = vpt_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```

## {.tabset .tabset-fade}
<h4>2.3 Heatmaps</h4>

<h4 id="DAAhm"> <a href="#back">2.3 Heatmaps </a></h4>

### By p-value
```{r fig.width=4, fig.height=8, echo=F, warning=F}
tar_read("HM_padj")
```

### By log2FoldChange
```{r fig.width=4, fig.height=8, echo=F, warning=F}
tar_read("HM_l2fc")
```


## {.tabset .tabset-fade}

<h3 id="GSEA"><a href="#back">3. GO enrichment analysis </a></h3>

All gene sets derived from Gene Ontology Database.
```{r echo=F, warning=F}
source("R/functions.R")
source("R/helper.R")

fe_results_up <- tar_read("fe.results.up")
fe_results_dn <- tar_read("fe.results.dn")

fe_results_dn$direction <- "DOWN"
fe_results_up$direction <- "UP" 
fe_results <- rbind(fe_results_up, fe_results_dn)
Table_title <- "Table of enriched GO termns (by interval)"
table_name <- "FE_GO_by_interval"
fe_results$source <- as.factor(fe_results$source)
dfs <- split.data.frame(fe_results, fe_results$source)

```

### GO BP
```{r echo=F, warning=F}
scrollX <- T

src <- "GO:BP"
Table_title <- "Table of enriched GO:BP termns (by interval)"
table_name <- "FE_GOBP_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```



### GO MF
```{r echo=F, warning=F}
src <- "GO:MF"
Table_title <- "Table of enriched GO:MF termns (by interval)"
table_name <- "FE_GOMF_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

### GO CC
```{r echo=F, warning=F}
src <- "GO:CC"
Table_title <- "Table of enriched GO:CC termns (by interval)"
table_name <- "FE_GOCC_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

### Reactome
```{r echo=F, warning=F}
src <- "REAC"
Table_title <- "Table of enriched REACTOME termns (by interval)"
table_name <- "FE_REAC_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

### Transcription_Factors
```{r echo=F, warning=F}
src <- "TF"
Table_title <- "Table of enriched TF termns (by interval)"
table_name <- "FE_TF_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```



### C2 gene sets

Gene sets in this collection are curated from various sources, including online pathway databases and the biomedical literature. Many sets are also contributed by individual domain experts. The gene set page for each gene set lists its source. The C2 collection is divided into the following two sub-collections: Chemical and genetic perturbations (CGP) and Canonical pathways (CP).

```{r echo=F, warning=F}

fe_results.c2_up <- tar_read("c2.fe.results.up")
fe_results.c2_dn <- tar_read("c2.fe.results.dn")
fe_results.c2_dn$direction <- "DOWN"
fe_results.c2_up$direction <- "UP" 
fe_results.c2 <- rbind(fe_results.c2_up, fe_results.c2_dn)
Table_title <- "Table of enriched C2 gene sets (by interval)"
table_name <- "FE_c2_by_interval"

fe_results.c2$source <- as.factor(fe_results.c2$source)
dfs <- split.data.frame(fe_results.c2, fe_results.c2$source)

DT::datatable(format_fe_results(fe_results.c2), rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

### Hallmark gene sets
```{r echo=F, warning=F}
fe_results.h_up <- tar_read("h.fe.results.up")
fe_results.h_dn <- tar_read("h.fe.results.dn")

fe_results.h_dn$direction <- "DOWN"
fe_results.h_up$direction <- "UP" 
fe_results.h <- rbind(fe_results.h_up, fe_results.h_dn)
Table_title <- "Table of enriched hallmark gene sets (by interval)"
table_name <- "FE_hallmark_by_interval"
DT::datatable(format_fe_results(fe_results.h), rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```


## {.tabset .tabset-fade}

<h3 id="Integration"> <a href="#back">4. Omics Integration </a></h3>

<h4>4.1 ATAC vs. Transcriptomics </h4>
```{r echo=F}
expression_data <- tar_read(DEA)
proteomics_data
read.table("results/proteomics/proteomics.csv", header=T) %>% rename(Symbol=SYMBOL, log2FoldChange=logFC, pvalue=p.val, padj=p.adj)
```


<h5> Tower Plot: Differential accessibility of top 25 up- and down-regulated genes </h5>

```{r fig.width=15, fig.height=6, warning=F, echo=F}
TP <- tar_read(TowerPlot_25)
pdf(file = "paper/plots/towerplot_top25_on_both_sides.pdf", width = 15, height = 6)
TP
dev.off()

```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",tp_p, sep="")),
  output_name = tp_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```


<h5> Correlation Plot </h5>

### All
```{r fig.width=10, fig.height=10, echo=F}
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 1, lfc2=0, padj2=1, es.cut = .56)

pdf(file = "paper/plots/correlation.plot.pdf", width = 10, height = 10)
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 1, lfc2=0, padj2=1, es.cut = .56)
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",cp_all_p, sep="")),
  output_name = cp_all_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)



```

### Significant differential accessibility
```{r fig.width=10, fig.height=10, echo=F}
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = .05, lfc2=0, padj2=1, es.cut = .56)

pdf(file = "paper/plots/correlation.plot.DA.significant.pdf", width = 10, height = 10)
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = .05, lfc2=0, padj2=1, es.cut = .56)
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",cp_das_p, sep="")),
  output_name = cp_das_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```

### Significant differential expression
```{r fig.width=10, fig.height=10, echo=F}
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 1, lfc2=0, padj2=.05, es.cut = .56)

pdf(file = "paper/plots/correlation.plot.DE.significant.pdf", width = 10, height = 10)
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 1, lfc2=0, padj2=.05, es.cut = .56)
dev.off()

```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",cp_des_p, sep="")),
  output_name = cp_des_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```

### Both significant
```{r fig.width=10, fig.height=10, echo=F, warning=F}
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 0.05, lfc2=0, padj2=.05, es.cut = .56, include.symbols = T)

pdf(file = "paper/plots/correlation.plot.both.significant.pdf", width = 10, height = 10)
correlation_plot(DAA = tar_read(results), DEA = tar_read(DEA), 
                 lfc1 = 0, padj1 = 0.05, lfc2=0, padj2=.05, es.cut = .56, include.symbols = T)
dev.off()
```
```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",cp_bos_p, sep="")),
  output_name = cp_bos_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```



## {.tabset .tabset-fade}

<h3 id="MotifanalysisFootprinting"> <a href="#back">5. Motif matching and footprinting </a></h3>


## {.tabset .tabset-fade}
<h4 id="MotifanalysisFootprintingma"><a href="#back"> 5.1 Motif-based analysis </a></h4>


<h5>Table</h5>
```{r echo=F, warning=F}

library(JASPAR2020)
library(TFBSTools)

motifanalysis_dir <- "results/motif.analysis/"
roi <- paste(motifanalysis_dir, "/enrichment/roi/",sep="")
match <- paste(motifanalysis_dir, "/match/",sep="")

fulltest_statistics_file <- paste(roi, "/fulltest_statistics.txt",sep="")
bed_file <- paste(match, "/roi_mpbs.bed",sep="")
ft_statsitics_df <- fulltest_statistics_file %>% read.table(header = T, sep = "\t")
Table_title <- "Motif enrichment analysis"
table_name <- "Motif_enrichment_analysis"




db <- file.path(system.file("extdata", package="JASPAR2020"), 
                    "JASPAR2020.sqlite")

ft_statsitics_df <- ft_statsitics_df %>% tibble::add_column(ID=gsub(pattern = "[[:punct:]][1-9]+[[:punct:]].*", replacement = "", .$FACTOR), .before = "FACTOR")

getFamily <- function(ID){
  return(`$`(`@`(TFBSTools::getMatrixByID(JASPAR2020, ID = ID),tags),family))
}
getFamily <-  Vectorize(function(ID) `$`(`@`(TFBSTools::getMatrixByID(JASPAR2020, ID = ID),tags),family) %>% unlist(),
                    vectorize.args = c("ID"))


ft_statsitics_df$FAMILY <- NULL
ft_statsitics_df <- ft_statsitics_df %>% tibble::add_column(FAMILY=getFamily(.$ID) %>% as.character(), .before = "P.VALUE")
ft_statsitics_df <- ft_statsitics_df %>% tibble::add_column(
  FoldChange=(as.double(gsub(pattern = "%",replacement = "",.$FREQ)) /
                      as.double(gsub(pattern = "%", replacement = "", .$BACK.FREQ.))), .before = "P.VALUE")
ft_statsitics_df <- ft_statsitics_df %>% tibble::add_column(GOI=if_else(grepl(pattern = "(Jun|Fos)",.$FAMILY), T, F), .before="FoldChange")

ft_statsitics_df <- ft_statsitics_df %>% add_column(Symbol=.$FACTOR)
ft_statsitics_df <- ft_statsitics_df %>% add_column(log2FoldChange=log2(.$FoldChange))
ft_statsitics_df <- ft_statsitics_df %>% add_column(padj=.$CORR.P.VALUE)

#ft_statsitics_df$GENES <- gsub(patter=",", replacement = " ", ft_statsitics_df$GENES)

DT::datatable(ft_statsitics_df[,1:13], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=T,     scroller=T,
    fixedColumns = TRUE,                                                                                                                     
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = T, pageLength = 10))

```


<h5>One-sided Volcano of enriched motifs</h5>

A = number of input regions with at least 1 occurrence of that particular transcription factor.
B = number of input regions with no occurrence of that particular transcription factor.
C = number of random background regions with at least 1 occurrence of that particular transcription factor.
D = number of random background regions with no occurrence of that particular transcription factor.
Here: foreground frequency ist (A/(A+B)), background frequency is (C/(C+D))

```{r fig.width=10, fig.height=7, echo=F, warning=F}
vpmp_table <- 
  ft_statsitics_df %>% 
  tibble::add_column(color=if_else(.$padj > 0.05 | .$log2FoldChange < .125,
                                   "lightgrey",
                                   if_else(grepl("(Fos-related factors)|(Jun-related factors)", .$FAMILY),
                                           "#d50054",
                                           "black")), 
                     .before="ID")

pdf(file="paper/plots/volcano.plot.motifs.pdf", width=10, height=7)

fc <- 1.15
plot_Volcano(DEG.table = vpmp_table, selected.genes = vpmp_table %>% filter(GOI==T & padj <= 0.05) %>% select(Symbol)%>% unlist() %>% unname(), 
                 colors.selected.genes = c("#d50054"),  l2fc =log2(fc), padj = 0.05,
                 xlab="log2(FC) of Motif Enrichment - KO vs. Background",
                 pt.size = 2, point.colors = vp.point.colors, repel=F,
                 pt.border.color = "lightgrey",
                 ylim=c(0, 128), xlim=c(-0,.5),subtitle = "JUN- or/and FOS-related motifs in red",
                 title= "One-sided Volcano of enriched motifs", title.color = "darkblue") + annotate(geom="text", x=log2(fc), y=50, label=fc, color="red")
dev.off()
```





```{r echo=F}
downloadthis::download_file(
  path = (paste("paper/plots/",vpm_p, sep="")),
  output_name = vpm_p,
  button_label = "Download pdf file",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-save",
  self_contained = T
)
```




## {.tabset .tabset-fade}

<h4 id="MotifanalysisFootprintingfp"><a href="#back"> 5.2 Footprinting </a></h4>
```{r echo=F}

footprinting_dir <- "results/footprinting/"
diff_footprints_dir <- paste(footprinting_dir, "/4_differential_footprints/LynKO_LynWT/", sep="")

activity_plot <- read.table(file = paste(diff_footprints_dir, "/differential_statistics.txt", sep=""), header = T, sep = "\t")


DT::datatable(activity_plot, rownames = FALSE, filter = 'top', 
                                       caption = "Footprinting analysis", extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip',     
                                                      scroller=T, fixedColumns = F,                                                                                                                     
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = T, pageLength = 10))


```

<h5>Summary</h5>

### Combined
```{r echo=F, fig.width=20, fig.height=7, warning=F}  
library(tidyr)
library(ggpubr)
footprinting.plot.norm <- footprinting_plot(activity_df=activity_plot, repel="label", repel.text.color="black", repel.text.size = 3, selected.genes = c(""), pt.colors=c("red", "grey", "steelblue"), repel.max.overlaps = 2000000, pt.sizes = c(2,2,2,2), title="Activity score distribution",  title.color = "#000080", title.face = "plain")[[1]]

footprinting.plot.volcano <- footprinting_plot(activity_df=activity_plot, activity = "Z_score", log10.second.variable=T, pt.colors=c("red", "grey", "steelblue"), second.variable = "P_values", selected.genes = c(""), ylim=c(-5,5), pt.sizes = c(2,0.5,2,2), ylab = "KO <-- z-score of Activity Score --> WT", xlab= "-log10(adjusted p-value) [significance]" ,sign.hline = "both", sign.hcut = 1, sign.vcut = -log10(0.05), sign.vline = "one", coord_flip = T, title="Activity score vs. Adjusted p-Value", title.color = "#000080", title.face = "plain")[[1]]

ggplotify::as.ggplot(cowplot::plot_grid(cowplot::plot_grid(footprinting.plot.norm, ggplot()+theme_void(), ncol=1, rel_heights = c(40,1)), footprinting.plot.volcano, ncol=2)) + ggtitle("Footprinting analysis - Summary") + theme(plot.title = element_text(hjust= 0.5, face = "bold", color="#800080", size=20))
```



### Activity Plot
Activity plot showing the distribution of z-transformend activity scores of the 
underlying motifs (y-axis) as computed by HINT-ATAC. On the x-axis the points are
distributed randomly using the base R sample function with replace=T and setting 
seed=123. Motifs with differential footprinting (p-Value <=0.05 and |z-Score| >= 1) 
are highlighted in red (enriched in KO) or in blue (enriched in WT). 
```{r echo=F, fig.width=10, fig.height=7, warning=F}  
footprinting.plot.norm
pdf(file = "paper/plots/footprinting.activity.plot.pdf", width = 10, height = 7)
footprinting.plot.norm
dev.off()
```

### Volcano-Like Plot (Z-score vs. Adjusted p-Value)
Volcano plot showing the relationship between z-transformed activity scores as 
computed by HINT-ATAC and the corresponding p-values. Motifs with differential
footprinting (p-Value <= 0.05 and |z-Score| >= 1) are highlighted in red 
(enriched in KO) or in blue (enrichedin WT). 

```{r echo=F, fig.width=10, fig.height=7, warning=F}  
footprinting.plot.volcano

pdf(file = "paper/plots/footprinting.plot.volcano.pdf", width = 10, height = 7)
footprinting.plot.volcano
dev.off()
```

### Volcano-Like Plot (log2FC of Protection Score vs. Adjusted p-Value)
Volcano plot showing the relationship between log2 fold change of protection
score (WT vs. KO) and the corresponding p-values. Motifs with top differential
footprinting (p-Value <= 0.05 and |l2fc| >= .1) are highlighted in red 
(enriched in KO) or in blue (enrichedin WT). 
```{r fig.width=8, fig.height=8, echo=F, warning=F}

ext.activity.plot <- activity_plot %>% add_column(log2FC=log2(.$Protection_Score_LynWT/.$Protection_Score_LynKO))

footprinting_plot(activity_df=ext.activity.plot, activity = "log2FC", log10.second.variable=T, pt.colors=c("red", "grey", "steelblue"), second.variable = "P_values", selected.genes = c(""), ylim=c(-1,1), pt.sizes = c(2,0.5,2,2), ylab = "KO <-- Log2 of Protection Score --> WT", xlab= "-log10(adjusted p-value) [significance]" ,sign.hline = "both", sign.hcut = .1, sign.vcut = -log10(0.05), sign.vline = "one", coord_flip = T, title="Log2 of Protection Score vs. Adjusted p-Value", title.color = "#000080", title.face = "plain")[[1]]

```
### Volcano-Like Plot (log2FC of Tag Counts vs. Adjusted p-Value)
Volcano plot showing the relationship between log2 fold change of tag counts
(WT vs. KO) and the corresponding p-values. Motifs with top differential
footprinting (p-Value <= 0.05 and |l2fc| >= .1) are highlighted in red 
(enriched in KO) or in blue (enrichedin WT). 
```{r fig.width=8, fig.height=8, echo=F, warning=F}

ext.activity.plot <- activity_plot %>% add_column(log2FC=log2(.$TC_LynWT/.$TC_LynKO))

footprinting_plot(activity_df=ext.activity.plot, activity = "log2FC", log10.second.variable=T, pt.colors=c("red", "grey", "steelblue"), second.variable = "P_values", selected.genes = c(""), ylim=c(-.5,.5), pt.sizes = c(2,0.5,2,2), ylab = "KO <-- Log2 of Tag Counts --> WT", xlab= "-log10(adjusted p-value) [significance]" ,sign.hline = "both", sign.hcut = .1, sign.vcut = -log10(0.05), sign.vline = "one", coord_flip = T, title="Log2 of Tag Counts vs. Adjusted p-Value", title.color = "#000080", title.face = "plain")[[1]]

```


## {.tabset .tabset-fade}

<h5>LinePlots (AP-1 subunits)  </h5>

```{r echo=F, fig.width=18, fig.height=18}
lineplots_dir <- "results/footprinting/4_differential_footprints/LynKO_LynWT/Lineplots/"
lineplots_file <- list.files(path = lineplots_dir, pattern = "*txt", full.names = T)
lineplots_names <- list.files(path = lineplots_dir, pattern = "*txt", full.names = F)
roi_lp_data_file_path <- as.data.frame(lineplots_file) %>% filter(grepl("NFKB|STAT1|JUN|FOS|(^ATF)",lineplots_file)) %>% as.list() %>% unlist()
roi_lp_data_names <- as.data.frame(lineplots_names) %>% filter(grepl("NFKB|THBS1|STAT1|JUN|FOS|(^ATF)",lineplots_file)) %>% as.list() %>% unlist() %>% gsub(pattern = "\\.txt", replacement = "")
lineplots <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  df <- read.table(roi_lp_data_file_path[r], header = T, sep="\t")
  df$id <- -100:99
  df <- df %>% tidyr::gather(key="Condition", value="Coverage", 1:2)
  lineplots[[r]] <- ggplot(df, aes(x=id, y=Coverage, colour=Condition, linetype=Condition)) + 
                    geom_line() + theme_pubr() + 
                    ylab("Coverage") + xlab("Position around footprint") + 
                    ggtitle(roi_lp_data_names[r]) + theme(plot.title = element_text(hjust=0.5, face="bold")) + 
                    scale_color_manual(values=c("grey", "red")) + scale_linetype_manual(values=c("solid", "solid"))
}

lines <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  
  very.strongly.sign.motifs <- activity_plot[activity_plot$P_values<0.001,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  strongly.sign.motifs <- activity_plot[activity_plot$P_values<0.01,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  sign.motifs <- activity_plot[activity_plot$P_values<0.05,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  weakly.sign.motifs <- activity_plot[activity_plot$P_values<0.1,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  
  all.motifs <- activity_plot$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  activity_plot$Motif.2 <- all.motifs
  
  standard.titles <- paste(roi_lp_data_names[r], " (",round(activity_plot[activity_plot$Motif.2 %in% c(roi_lp_data_names[r]),]$P_values, 2),")",sep="")
  
  sign.titles <- paste(standard.titles," *",sep="")
  strongly.sign.titles <- paste(standard.titles," **",sep="")
  very.strongly.sign.titles <- paste(standard.titles," ***",sep="")
  weak.sign.titles <- paste(standard.titles," +",sep="")
 
  if(F){ 
  writeLines( c(paste("### ",ifelse(roi_lp_data_names[r] %in% very.strongly.sign.motifs, 
                                    very.strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% strongly.sign.motifs, 
                                    strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% sign.motifs, 
                                           sign.titles,
                                           ifelse(roi_lp_data_names[r] %in% weakly.sign.motifs, 
                                                  weak.sign.titles,
                                                  standard.titles)))
                                    ),
                      sep=""), 
               "```{r echo=F}", 
               paste("lineplots[[",r,"]]",sep=""),
               "```",
               "")
  )
  }
 
}
```

### MA0099.3.FOS::JUN (0.88)
```{r echo=F}
lineplots[[1]]
```

### MA0105.4.NFKB1 (0.75)
```{r echo=F}
lineplots[[2]]
```

### MA0137.3.STAT1 (0.55)
```{r echo=F}
lineplots[[3]]
```

### MA0462.2.BATF::JUN (0.69)
```{r echo=F}
lineplots[[4]]
```

### MA0476.1.FOS (0.23)
```{r echo=F}
lineplots[[5]]
```

### MA0477.2.FOSL1 (0.62)
```{r echo=F}
lineplots[[6]]
```

### MA0478.1.FOSL2 (0.91)
```{r echo=F}
lineplots[[7]]
```

### MA0488.1.JUN (0.08) +
```{r echo=F}
lineplots[[8]]
```

### MA0489.1.JUN_var.2 (0.72)
```{r echo=F}
lineplots[[9]]
```

### MA0490.2.JUNB (0.02) *
```{r echo=F}
lineplots[[10]]
```

### MA0491.2.JUND (0.03) *
```{r echo=F}
lineplots[[11]]
```

### MA0492.1.JUND_var.2 (0.04) *
```{r echo=F}
lineplots[[12]]
```

### MA0517.1.STAT1::STAT2 (0.11)
```{r echo=F}
lineplots[[13]]
```

### MA0778.1.NFKB2 (0.42)
```{r echo=F}
lineplots[[14]]
```

### MA1126.1.FOS::JUN_var.2 (0.04) *
```{r echo=F}
lineplots[[15]]
```

### MA1127.1.FOSB::JUN (0.02) *
```{r echo=F}
lineplots[[16]]
```

### MA1128.1.FOSL1::JUN (0.09) +
```{r echo=F}
lineplots[[17]]
```

### MA1129.1.FOSL1::JUN_var.2 (0.01) **
```{r echo=F}
lineplots[[18]]
```

### MA1130.1.FOSL2::JUN (0.88)
```{r echo=F}
lineplots[[19]]
```

### MA1131.1.FOSL2::JUN_var.2 (0.08) +
```{r echo=F}
lineplots[[20]]
```

### MA1132.1.JUN::JUNB (0.73)
```{r echo=F}
lineplots[[21]]
```

### MA1133.1.JUN::JUNB_var.2 (0.06) +
```{r echo=F}
lineplots[[22]]
```

### MA1134.1.FOS::JUNB (0.84)
```{r echo=F}
lineplots[[23]]
```

### MA1135.1.FOSB::JUNB (0.63)
```{r echo=F}
lineplots[[24]]
```

### MA1136.1.FOSB::JUNB_var.2 (0.006) ***
```{r echo=F}
lineplots[[25]]
```

### MA1137.1.FOSL1::JUNB (0.29)
```{r echo=F}
lineplots[[26]]
```

### MA1138.1.FOSL2::JUNB (0.29)
```{r echo=F}
lineplots[[27]]
```

### MA1139.1.FOSL2::JUNB_var.2 (0.01) **
```{r echo=F}
lineplots[[28]]
```

### MA1140.2.JUNB_var.2 (0.49)
```{r echo=F}
lineplots[[29]]
```

### MA1141.1.FOS::JUND (0.38)
```{r echo=F}
lineplots[[30]]
```

### MA1142.1.FOSL1::JUND (0.01) **
```{r echo=F}
lineplots[[31]]
```

### MA1143.1.FOSL1::JUND_var.2 (0.01) **
```{r echo=F}
lineplots[[32]]
```

### MA1145.1.FOSL2::JUND_var.2 (0.02) *
```{r echo=F}
lineplots[[33]]
```
## {.tabset .tabset-fade}

<h5>LinePlots (differentially binding TF)  </h5>

```{r echo=F}
"MA0746.2.SP3|MA0888.1.EVX2|MA0862.1.GMEB2|MA0604.1.Atf1|MA0060.3.NFYA|MA0844.1.XBP1|MA1564.1.SP9|MA1475.1.CREB3L4(var.2)|MA0840.1.Creb5|MA1142.1.FOSL1::JUND|MA0834.1.ATF7|MA0837.1.CEBPE|MA1143.1.FOSL1::JUND(var.2)|MA0886.1.EMX2|MA1634.1.BATF|MA1125.1.ZNF384|MA0685.1.SP4|MA1683.1.FOXA3|MA1126.1.FOS::JUN(var.2)|MA0835.2.BATF3|MA1145.1.FOSL2::JUND(var.2)|MA0490.2.JUNB|MA0516.2.SP2|MA0638.1.CREB3|MA0492.1.JUND(var.2)|MA1136.1.FOSB::JUNB(var.2)|MA1560.1.SOHLH2|MA0491.2.JUND|MA0893.2.GSX2|MA0502.2.NFYB|MA1470.1.BACH2(var.2)|MA0912.2.HOXD3|MA0079.4.SP1|MA0050.2.IRF1|MA1632.1.ATF2|MA1127.1.FOSB::JUN|MA0740.1.KLF14|MA0464.2.BHLHE40|MA0609.2.CREM|MA1557.1.SMAD5|MA1129.1.FOSL1::JUN(var.2)|MA0755.1.CUX2|MA1139.1.FOSL2::JUNB(var.2)"


```


```{r echo=F, fig.width=18, fig.height=18}
lineplots_dir <- "~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/4_differential_footprints/LynKO_LynWT/Lineplots/"
diff_statistics <- read.table("~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/4_differential_footprints/LynKO_LynWT/differential_statistics.txt", sep = "\t", header = T)
diff_statistics <- diff_statistics %>% filter(P_values <= 0.05)
lineplots_file <- list.files(path = lineplots_dir, pattern = "*txt", full.names = T)
lineplots_names <- list.files(path = lineplots_dir, pattern = "*txt", full.names = F)



roi_lp_data_file_path <- as.data.frame(lineplots_file) %>% filter(grepl(paste("(",paste(diff_statistics$Motif,collapse = ")|("),")",sep=""),lineplots_file)) %>% as.list() %>% unlist()
roi_lp_data_names <- as.data.frame(lineplots_names) %>% filter(grepl(paste("(",paste(diff_statistics$Motif,collapse = ")|("),")",sep=""),lineplots_file)) %>% as.list() %>% unlist() %>% gsub(pattern = "\\.txt", replacement = "")
lineplots <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  df <- read.table(roi_lp_data_file_path[r], header = T, sep="\t")
  df$id <- -100:99
  df <- df %>% tidyr::gather(key="Condition", value="Coverage", 1:2)
  lineplots[[r]] <- ggplot(df, aes(x=id, y=Coverage, colour=Condition, linetype=Condition)) + 
                    geom_line() + theme_pubr() + 
                    ylab("Coverage") + xlab("Position around footprint") + 
                    ggtitle(roi_lp_data_names[r]) + theme(plot.title = element_text(hjust=0.5, face="bold")) + 
                    scale_color_manual(values=c("grey", "red")) + scale_linetype_manual(values=c("solid", "solid"))
}

lines <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  
  very.strongly.sign.motifs <-  activity_plot[activity_plot$P_values<0.001,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
       strongly.sign.motifs <-  activity_plot[activity_plot$P_values<0.01,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
                sign.motifs <-  activity_plot[activity_plot$P_values<0.05,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
         weakly.sign.motifs <-  activity_plot[activity_plot$P_values<0.1,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
                 all.motifs <-  activity_plot$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
  activity_plot$Motif.2 <- all.motifs
  standard.titles <- paste(roi_lp_data_names[r]," ",
                           "(",
                           round(activity_plot[activity_plot$Motif.2 %in% c(roi_lp_data_names[r]),]$P_values, 2),
                           ")",
                           sep="")
  sign.titles <- paste(standard.titles," *",sep="")
  strongly.sign.titles <- paste(standard.titles," **",sep="")
  very.strongly.sign.titles <- paste(standard.titles," ***",sep="")
  weak.sign.titles <- paste(standard.titles," +",sep="")
  
  writeLines( c(paste("### ",ifelse(roi_lp_data_names[r] %in% very.strongly.sign.motifs, 
                                    very.strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% strongly.sign.motifs,
                                           strongly.sign.titles,
                                           ifelse(roi_lp_data_names[r] %in% sign.motifs,
                                                  sign.titles,
                                                  ifelse(roi_lp_data_names[r] %in% weakly.sign.motifs,
                                                         weak.sign.titles,
                                                         standard.titles)))),sep=""), 
                     "```{r echo=F}", 
                     paste("lineplots[[",r,"]]",sep=""),
                     "```",
                     ""))
}
```


### MA0050.2.IRF1 (0) ***
```{r echo=F}
lineplots[[1]]
```

### MA0060.3.NFYA (0) **
```{r echo=F}
lineplots[[2]]
```

### MA0079.4.SP1 (0.01) *
```{r echo=F}
lineplots[[3]]
```

### MA0464.2.BHLHE40 (0.02) *
```{r echo=F}
lineplots[[4]]
```

### MA0490.2.JUNB (0.02) *
```{r echo=F}
lineplots[[5]]
```

### MA0491.2.JUND (0.03) *
```{r echo=F}
lineplots[[6]]
```

### MA0502.2.NFYB (0) ***
```{r echo=F}
lineplots[[7]]
```

### MA0516.2.SP2 (0.01) *
```{r echo=F}
lineplots[[8]]
```

### MA0604.1.Atf1 (0.02) *
```{r echo=F}
lineplots[[9]]
```

### MA0609.2.CREM (0.04) *
```{r echo=F}
lineplots[[10]]
```

### MA0638.1.CREB3 (0.03) *
```{r echo=F}
lineplots[[11]]
```

### MA0685.1.SP4 (0) **
```{r echo=F}
lineplots[[12]]
```

### MA0740.1.KLF14 (0) **
```{r echo=F}
lineplots[[13]]
```

### MA0746.2.SP3 (0.02) *
```{r echo=F}
lineplots[[14]]
```

### MA0755.1.CUX2 (0.01) **
```{r echo=F}
lineplots[[15]]
```

### MA0834.1.ATF7 (0.04) *
```{r echo=F}
lineplots[[16]]
```

### MA0835.2.BATF3 (0.02) *
```{r echo=F}
lineplots[[17]]
```

### MA0837.1.CEBPE (0.02) *
```{r echo=F}
lineplots[[18]]
```

### MA0840.1.Creb5 (0.04) *
```{r echo=F}
lineplots[[19]]
```

### MA0844.1.XBP1 (0.01) **
```{r echo=F}
lineplots[[20]]
```

### MA0862.1.GMEB2 (0.01) **
```{r echo=F}
lineplots[[21]]
```

### MA0886.1.EMX2 (0.03) *
```{r echo=F}
lineplots[[22]]
```

### MA0888.1.EVX2 (0) ***
```{r echo=F}
lineplots[[23]]
```

### MA0893.2.GSX2 (0) **
```{r echo=F}
lineplots[[24]]
```

### MA0912.2.HOXD3 (0.03) *
```{r echo=F}
lineplots[[25]]
```

### MA1125.1.ZNF384 (0) **
```{r echo=F}
lineplots[[26]]
```

### MA1127.1.FOSB::JUN (0.02) *
```{r echo=F}
lineplots[[27]]
```

### MA1142.1.FOSL1::JUND (0.01) **
```{r echo=F}
lineplots[[28]]
```

### MA1557.1.SMAD5 (0.04) *
```{r echo=F}
lineplots[[29]]
```

### MA1560.1.SOHLH2 (0) ***
```{r echo=F}
lineplots[[30]]
```

### MA1564.1.SP9 (0.04) *
```{r echo=F}
lineplots[[31]]
```

### MA1632.1.ATF2 (0.03) *
```{r echo=F}
lineplots[[32]]
```

### MA1634.1.BATF (0) ***
```{r echo=F}
lineplots[[33]]
```

### MA1683.1.FOXA3 (0) ***
```{r echo=F}
lineplots[[34]]
```

### Author and session Information
<h4>Analyzed by</h4>
Dr. Ali T. Abdallah, Dipl. Inf.</br>
Bioinformatics Facility (Coordinator)</br>
University Hospital of Cologne - CECAD Research Center</br>
Joseph-Stelzmann-Str. 26, 52074 Cologne, Germany</br>
Phone +49 221 478 84017</br>

<h4>Detailed Session Info</h4>
```{r echo=FALSE}
sessionInfo()
```