---
title: 'ATAC-seq-analysis: Reprogramming of fibroblasts'
author:
  - name: Ali T. Abdallah ^[CECAD Research Center - University of Cologne, ali.abdallah@rwth-aachen.de]
fontsize: 15pt
date: "`r Sys.Date()`"
output: 
  html_notebook:
  toc: yes
  highlight: kate
  theme: yeti
  css: "style/theme.css"
---

<style>
.main-container{
    min-width: 1800px;!important
    margin-left: auto;!important
    margin-right: auto;!important
    font-size: 15px;!important
}

#toc_container {
    background: #f9f9f9 none repeat scroll 0 0;
    border: 0px solid #aaa;
    display: table;
    font-size: 100%;
    margin-bottom: 1em;
    padding: 10px;
    width: auto;
}

.toc_title {
    font-weight: 700;
    text-align: left;
}

#toc_container li, #toc_container ol, #toc_container ol li{
    list-style: outside none none !important;
}

h3{
  color: #800000;
  font-weight: bold;
}

h4{
  color: #000080;
  font-weight: bold;
}

h4 a{
  color: #000080;
}

h5{
  color: #008080;
  font-weight: bold
}

h5 a{
  color: #008080;
}

</style>


<div id="toc_container">
<p class="toc_title" id="back"><h4 style="color:#800000"><b>Contents</b></h4></p>
<ol class="toc_list" style="font-size:15px">
  
  <li><a href="#QC"><b>1. Quality Control</b></a>
      <ol class="toc_list">
      <li><a href="#QCfld">1.1 Fragement length distribution </a></li>
      <li><a href="#QCpcash">1.2 PCA analysis and similarity heatmap </a></li>
    </ol>
  </li>
  
  <li><a href="#DAA"><b>2. Differential Accessibility analysis</b></a> 
    <ol class="toc_list">
      <li><a href="#DAAt">2.1 Tables</a></li>
      <li><a href="#DAAvp">2.2 Volcano plots</a></li>
      <li><a href="#DAAhm">2.3 Heatmaps</a></li>
    </ol>
  </li>

  <li><a href="#GSEA"><b>3. GO enrichment analysis </b></a>
    <ol class="toc_list">
      <li><a href="#GSEAt">3.1 Tables</a></li>
      <li><a href="#GSEAv">3.2 Visualization</a></li>
    </ol>
  </li>
  
  <li><a href="#Integration"><b>4. Omics Integration</b></a>
    <ol class="toc_list">
      <li><a href="#Integrationg">4.1 Correlation analysis on gene level </a> </li>
      <li><a href="#Integrationgs">4.2 Correlation analysis on gene set level</a></li>
    </ol>
  </li>
  
  <li><a href="#MotifanalysisFootprinting"><b>5. Motif matching and footprinting </b></a>
      <ol class="toc_list">
      <li><a href="#MotifanalysisFootprintingvp">5.1 Volcano plots of TF targets </a></li>
      <li><a href="#MotifanalysisFootprintingma">5.2 Motif analysis </a></li>
      <li><a href="#MotifanalysisFootprintingfp">5.3 footprinting </a></li>
      </ol>
  </li>
</ol>
</div>



## {.tabset .tabset-fade}


```{r setup, echo=F}
source("./R/functions.R")
source("./R/NGSTools.R")
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(targets)
library(ggplot2)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
txdb$user_seqlevels <- gsub(pattern="chr", replacement = "", txdb$user_seqlevels)
tar_make()
print("")
```


<h3 id="QC"><a href="#back"> 1. Quality Control </a></h3>

<h4 id="QCfld"> <a href="#back"> 1.1 Fragment length distribution </a></h4>

<h4 id="QCpcash"> <a href="#back">  1.2 PCA and similarity heatmap </a></h4>

### PCA and similarity heatmap
```{r fig.width=12, fig.height=5, echo=F}
colData <- tar_read(colData)
ggplotify::as.ggplot(cowplot::plot_grid(tar_read(PCA_plot), tar_read(SIM_hm), ncol=2)) + #
  ggtitle(paste(unique(colData$condition)[1], " vs. ", unique(colData$condition)[2], sep="")) + 
  theme(plot.title=element_text(face = "bold", hjust=0.5, colour = "#000080"))

```

### Meta.Data
```{r echo=F}
Table_title <- "Metadata of Lyn.KO experiment"
table_name <- "Lyn.Ko.Meta.Data"
DT::datatable(colData, rownames = T, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))
```

## {.tabset .tabset-fade}
<h3>2. Differential accessibility analysis </h3>

<h4>2.1 Tables</h4>

### Promoter
```{r echo=F, warning=F}
Table_title <- "Table of differentially accessible genes (by interval - promoter region)"
table_name <- "Diff_accessible_genes_by_interval_promoter"
sign.results <- tar_read(sign.results)
DT::datatable(sign.results[sign.results$reduced.annotation %in% c("Promoter"), ], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))

```

### All
```{r echo=F, warning=F}
Table_title <- "Table of differentially accessible genes (by interval)"
table_name <- "Diff_accessible_genes_by_interval"
DT::datatable(sign.results, rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=F,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20))




```

```{r echo=F}
generate_motif_dataframe_bed <- function(annotatedPeaks.df, results){
  sr <- results %>% dplyr::filter(Geneid %in% annotatedPeaks.df$Geneid) %>%  dplyr::select(Geneid, Symbol, log2FoldChange, padj) %>% na.omit()
  motif.bed.file <- annotatedPeaks.df %>% 
                    dplyr::filter(Geneid %in% results$Geneid) %>%  
                    dplyr::select(seqnames, start, end, Geneid, Symbol) %>% 
                    na.omit() %>% 
                    merge(sr, by=c("Geneid", "Symbol")) %>%
                    relocate(Geneid, Symbol, .after=end) %>% 
                    arrange(seqnames, start, end) 
  return(motif.bed.file)
}

d1 <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)

target_dir <- "~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedReplicate/indir/motif.analysis/"
roi <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)
bg <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = results)
write.table(roi, file=paste(target_dir,"/","roi.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
write.table(bg, file=paste(target_dir,"/","bg.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
```


```{r echo=F}
if(F){
  generate_motif_dataframe_bed <- function(annotatedPeaks.df, dap.df, seq.colum.name="seqnames", 
                                           start.column.name="start", end.column.name="end",
                                           peak.column.name="Geneid", gene.column.name="Symbol", 
                                           l2fc.column.name="log2FoldChange", padj.column.name="padj"){
    
    sr <- dap.df %>% 
          dplyr::filter(get(peak.column.name) %in% annotatedPeaks.df[,peak.column.name]) %>% 
          dplyr::select(peak.column.name, gene.column.name, l2fc.column.name, padj.column.name) %>%
          na.omit()
    motif.bed.file <- annotatedPeaks.df %>% 
                      dplyr::filter(get(peak.column.name) %in% dap.df[,peak.column.name]) %>%  
                      dplyr::select(seq.colum.name, start.column.name, end.column.name, peak.column.name, gene.column.name) %>% 
                      na.omit() %>% 
                      merge(sr, by=c(peak.column.name, gene.column.name)) %>%
                      relocate(peak.column.name, gene.column.name, .after=end.column.name) %>% 
                      arrange(get(seq.colum.name), get(start.column.name), get(end.column.name) )
    return(motif.bed.file)
  }
  
 d2 <-  generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, dap.df = sign.results)
   
  target_dir <- "~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedReplicate/indir/motif.analysis/"
  roi <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = sign.results)
  bg <- generate_motif_dataframe_bed(annotatedPeaks.df = annotatedPeaks.df, results = results)
  write.table(roi, file=paste(target_dir,"/","roi.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
  write.table(bg, file=paste(target_dir,"/","bg.bed", sep=""), col.names = F, row.names=F, quote=F, sep="\t")
}
```

## {.tabset .tabset-fade}

<h4>2.2 Volcano Plots</h4>

### Combined
```{r fig.height=10, fig.width=21, warning=F, echo=F}
vpi <- tar_read(VPI)
vpt <- tar_read(VPT)
vpg <- tar_read(VPG)
ggplotify::as.ggplot(plot_grid(ggplot()+theme_void(),plot_grid(vpi, vpt, ncol=2), rel_heights = c(1,20), ncol=1)) + ggtitle("Volcano plots at both levels (Peak, Transcript)", subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + theme(plot.title = element_text(face="bold", colour = "#000080", hjust = 0.5, size=18), plot.subtitle = element_text(face="plain", colour = "black", hjust = 0.5, size=12))


```

### Intervall
```{r fig.height=6, fig.width=7, warning=F, echo=F}
vpi+ ggtitle("Volcano plot at peak level", subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + theme(plot.title = element_text(face="bold", colour = "#000080", hjust = 0.5, size=18), plot.subtitle = element_text(face="plain", colour = "black", hjust = 0.5, size=12))
```
### Transcript
```{r fig.height=6, fig.width=7, warning=F, echo=F}
vpt+ ggtitle("Volcano plot at transcript level", subtitle = "Differential accessibility of Lyn.KO vs. Lyn.WT from ATAC-Seq data") + theme(plot.title = element_text(face="bold", colour = "#000080", hjust = 0.5, size=18), plot.subtitle = element_text(face="plain", colour = "black", hjust = 0.5, size=12))
```

## {.tabset .tabset-fade}
<h4>2.3 Heatmaps</h4>


### By p-value
```{r fig.width=4, fig.height=8, echo=F, warning=F}
tar_read("HM_padj")
```

### By log2FoldChange
```{r fig.width=4, fig.height=8, echo=F, warning=F}
tar_read("HM_l2fc")
```


## {.tabset .tabset-fade}

<h3>3. GO enrichment analysis </h3>

All gene sets derived from Gene Ontology Database.
```{r echo=F, warning=F}
source("R/functions.R")

de_results <- na.omit(sign.results[sign.results$log2FoldChange>0,])
de_results <- de_results[de_results$condition %in% c("Lyn.KO"),]
fe_results_up <- getFEResults(de_results = de_results, direction = "UP", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")
de_results <- na.omit(sign.results[sign.results$log2FoldChange<0,])
de_results <- de_results[de_results$condition %in% c("Lyn.WT"),]
fe_results_dn <- getFEResults(de_results = de_results, direction = "DOWN", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")

format_fe_results <- function(fe_results){
  fe_results <- fe_results[, c("source", "direction", "term_id", "term_name", "p_value", "term_size", "query_size", "intersection_size", "intersection", "precision", "recall")]
  colnames(fe_results) <-    c("source", "direction", "term_id", "term_name", "p_value", "term_size", "query_size", "number_of_shared_genes", "shared_genes", "precision", "recall")
  return(fe_results)
}


fe_results_dn$direction <- "down"
fe_results_up$direction <- "up" 
fe_results <- rbind(fe_results_up, fe_results_dn)
Table_title <- "Table of enriched GO termns (by interval)"
table_name <- "FE_GO_by_interval"
fe_results$source <- as.factor(fe_results$source)
dfs <- split.data.frame(fe_results, fe_results$source)

```

### GO BP
```{r echo=F, warning=F}
scrollX <- T

src <- "GO:BP"
Table_title <- "Table of enriched GO:BP termns (by interval)"
table_name <- "FE_GOBP_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```



### GO MF
```{r echo=F, warning=F}
src <- "GO:MF"
Table_title <- "Table of enriched GO:MF termns (by interval)"
table_name <- "FE_GOMF_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```

### GO CC
```{r echo=F, warning=F}
src <- "GO:CC"
Table_title <- "Table of enriched GO:CC termns (by interval)"
table_name <- "FE_GOCC_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```

### Reactome
```{r echo=F, warning=F}
src <- "REAC"
Table_title <- "Table of enriched REACTOME termns (by interval)"
table_name <- "FE_REAC_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```

### Transcription_Factors
```{r echo=F, warning=F}
src <- "TF"
Table_title <- "Table of enriched TF termns (by interval)"
table_name <- "FE_TF_by_interval"
dfs[[src]]$source <- as.factor(dfs[[src]]$source)
DT::datatable(format_fe_results(dfs[[src]])[,-1], rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```



### C2 gene sets

Gene sets in this collection are curated from various sources, including online pathway databases and the biomedical literature. Many sets are also contributed by individual domain experts. The gene set page for each gene set lists its source. The C2 collection is divided into the following two sub-collections: Chemical and genetic perturbations (CGP) and Canonical pathways (CP).

```{r echo=F, warning=F}
source("R/functions.R")
customDB.c2 <- upload_GMT_file(gmtfile = "../Data/c2.all.v7.5.1.symbols.gmt")

de_results.c2 <- na.omit(sign.results[sign.results$log2FoldChange>0,])
de_results.c2 <- de_results.c2[de_results.c2$condition %in% c("Lyn.KO"),]
fe_results.c2_up <- getFEResults(de_results = de_results.c2, DB = customDB.c2, direction = "UP", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")
de_results.c2 <- na.omit(sign.results[sign.results$log2FoldChange<0,])
de_results.c2 <- de_results.c2[de_results.c2$condition %in% c("Lyn.WT"),]
fe_results.c2_dn <- getFEResults(de_results = de_results.c2, DB = customDB.c2, direction = "DOWN", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")

format_fe_results.c2 <- function(fe_results.c2){
  fe_results.c2 <- fe_results.c2[, c("direction", "term_id", "p_value", "term_size", "query_size", "intersection_size", "precision", "recall", "intersection", "source")]
  return(fe_results.c2)
}

fe_results.c2_dn$direction <- "down"
fe_results.c2_up$direction <- "up" 
fe_results.c2 <- rbind(fe_results.c2_up, fe_results.c2_dn)
Table_title <- "Table of enriched C2 gene sets (by interval)"
table_name <- "FE_c2_by_interval"

fe_results.c2$source <- as.factor(fe_results.c2$source)
dfs <- split.data.frame(fe_results.c2, fe_results.c2$source)

DT::datatable(format_fe_results.c2(fe_results.c2), rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```

### Hallmark gene sets
```{r echo=F, warning=F}
source("R/functions.R")
customDB.hallmarks <-  upload_GMT_file(gmtfile = "../Data/h.all.v7.5.1.symbols.gmt")

de_results.h <- na.omit(sign.results[sign.results$log2FoldChange>0,])
de_results.h <- de_results.h[de_results.h$condition %in% c("Lyn.KO"),]
fe_results.h_up <- getFEResults(de_results = de_results.h, DB = customDB.hallmarks, direction = "UP", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")
de_results.h <- na.omit(sign.results[sign.results$log2FoldChange<0,])
de_results.h <- de_results.h[de_results.h$condition %in% c("Lyn.WT"),]
fe_results.h_dn <- getFEResults(de_results = de_results.h, DB = customDB.hallmarks, direction = "DOWN", ordered=F, organism = "hsapiens", gene.column.name = "Symbol")

format_fe_results.h <- function(fe_results.h){
  fe_results.h <- fe_results.h[, c("direction", "term_id", "p_value", "term_size", "query_size", "intersection_size", "precision", "recall", "intersection", "source")]
  return(fe_results.h)
}

fe_results.h_dn$direction <- "down"
fe_results.h_up$direction <- "up" 
fe_results.h <- rbind(fe_results.h_up, fe_results.h_dn)
Table_title <- "Table of enriched C2 gene sets (by interval)"
table_name <- "FE_hallmark_by_interval"
DT::datatable(format_fe_results.h(fe_results.h), rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=scrollX,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))
```


## {.tabset .tabset-fade}

<h3>4. Omics Integration </h3>

<h4>4.1 ATAC vs. Transcriptomics </h4>
```{r echo=F}

expression_data <- read.table(file="results/transcriptomics/transcriptomics.csv", sep="\t", header=T)

proteomics_data <- read.table(file="results/proteomics/proteomics.csv", sep="\t", header=T)
secretome_data <- read.table(file="results/secretomics/secretome.csv", sep="\t", header=T)

colnames(expression_data) <- c("Symbol", "Description", "log2FoldChange", "CI.L", "CI.R", "pvalue", "padj", "significant")
colnames(proteomics_data) <- c("Symbol", "ID", "log2FoldChange", "pvalue", "padj", "significant")
colnames(secretome_data) <- c("Symbol", "ID", "Log2FoldChange", "pvalue", "padj", "Location")


write.table(expression_data, file="~/Desktop/RTools/ATAC-Seq/data/transcriptomics.csv", quote=F, row.names = F, sep="\t")

#expression_data <- expression_data %>% add_column(Direction=if_else(.$log2FoldChange>=0, 1, -1), .before = "Symbol")
#expression_data$log2FoldChange.gw <- (expression_data$Direction)*expression_data$log2FoldChange
#expression_data$Direction <- as.factor(expression_data$Direction)
#expression_data <- expression_data %>% filter((log2FoldChange.gw) >= 1 & padj <= 0.05) %>% group_by(Direction) %>% dplyr::top_n(wt = (log2FoldChange.gw), n= 25) %>% arrange(Direction, log2FoldChange)
#expression_data <- expression_data %>% add_column(ID=1:nrow(expression_data), .before="Direction")
```


```{r fig.width=12, fig.height=6, echo=F}
res <- read.table("~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedLibrary/macs/narrowPeak/consensus/deseq2/KOvsWT/KOvsWT.mLb.clN.deseq2.results.txt", sep="\t", header=T)
apeaks <- read.table("~/Desktop/CHEOPS/projects/cecad/bioinformatics-core-facility/projects/2_primary.analysis/AG_Hallek/xxxxxx_vomstein_atacseq/results/bwa/mergedLibrary/macs/narrowPeak/consensus/consensus_peaks.mLb.clN.boolean.annotatePeaks.txt", sep="\t", header=T)
colnames(apeaks)[4] <- "Geneid"
res <- res %>% merge(apeaks, by="Geneid")
epi <- res %>% filter(res$Gene.Name %in% expression_data$Symbol) %>% filter(Distance.to.TSS < 100000) %>% arrange(-abs(log2FoldChange))
epi[epi$Gene.Name=="UCHL1",]
table(epi$Gene.Name)
corr_trans_epi[order(corr_trans_epi$log2FoldChange.x),]
```

<h5> Differential accessibility of top 25 up- and down-regulated genes </h5>

```{r fig.width=15, fig.height=6, warning=F, echo=F}

TowerPlot <- function(deg_data, dap_data, annotation.name="annotation", region.name="", gene.name="Symbol",
                          padj.trans=0.05, padj.atac=0.05, only.significant.peaks=F, color.border.sign.peaks="black",
                          l2fc.trans=1, color.border.unsign.peaks="darkgrey", color.region.of.interest="yellow", 
                          roi=c("Promoter","3' UTR"), gene.label.angle=90, point.shape=23, min.point.size=3, 
                          max.point.size=10, atac.regulation.color.low="steelblue",  roi.rev=F, ylim=c(-10,10), 
                          stack.inc=0.5, atac.regulation.color.high="darkred", x.lab="Top 50 Trans DEG", 
                          y.lab="Log2FC Trans", top_n=25, stroke.high=1.5, stroke.std=1, move.gene.name.down=1.2,
                          color.gene.name="black", order.by="padj", order.dir="desc", theme=theme_pubr(),
                          title="", title.face="bold", title.color="black", title.hjust=0.5,
                          legend.title.color="black", legend.title.face="bold", legend.title.background.fc="Log2FC ATAC",
                          legend.title.background.padj="-Log10 P-Value", legend.title.point.types="Peak type",
                          legend.background.padj.point.color="black", return=c("plot", "plot+data","data"),
                          legend.padj.point.size.percentiles=c(.05, .25, .5, .75, 1)){

  ### Combine and reshape data for downstream usage.
  if(is.null(deg_data$RegulationDirection)){
    deg_data <- deg_data %>% add_column(RegulationDirection=if_else(.$log2FoldChange>=0, 1, -1), .before = gene.name)
  }
  deg_data$log2FoldChange.gw <- abs(deg_data$log2FoldChange)
  deg_data$RegulationDirection <- as.factor(deg_data$RegulationDirection)
  deg_data <- deg_data %>%
              filter((log2FoldChange.gw) >= l2fc.trans & padj <= padj.trans) %>%
              group_by(RegulationDirection) %>%
              dplyr::top_n(wt = (log2FoldChange.gw), n= top_n) %>% 
              arrange(RegulationDirection, log2FoldChange)
  if(is.null(deg_data$ID)){
  deg_data <- deg_data %>%
              add_column(ID=1:nrow(deg_data), .before="RegulationDirection")
  }
  
    dap_data <- dap_data[dap_data$Symbol %in% deg_data$Symbol,]

  if(roi.rev){corr_trans_epi <- deg_data %>% merge(dap_data[!grepl(region.name,dap_data[,annotation.name]),], by=gene.name)}
  else{corr_trans_epi <- deg_data %>% merge(dap_data[grepl(region.name,dap_data[,annotation.name]),], by=gene.name)}
  

  print(corr_trans_epi)
  
  if(only.significant.peaks){ corr_trans_epi <- corr_trans_epi %>% filter(padj.y <= padj.atac)} 
  order.var <- ifelse(order.by=="l2fc", "log2FoldChange.y", ifelse(order.by=="padj", "padj.y", ""))
  corr_trans_epi$s.annotation <- gsub(x=corr_trans_epi$annotation, pattern = " \\(.*\\)", "")
  
  ### Compute position of peaks on each stack across y axis. Variable stac.inc define the density of the points on the
  ### the stack
  corr_trans_epi <- corr_trans_epi  %>% arrange(get(gene.name), abs(get(order.var))) %>% group_by(get(gene.name)) %>%
                    mutate(groupid = (cur_group_id())) %>%  mutate(gid = cumsum(groupid)/cur_group_id()) %>% ungroup() %>% 
                    add_column(position=(ifelse(.$gid==1,0,stack.inc*(.$gid-1))+.$log2FoldChange.x), .before="log2FoldChange.x")
  
  ### Only annotate first peak with gene name for each gene.
  corr_trans_epi <- corr_trans_epi %>% group_by(groupid) %>% mutate(groupid=cur_group_id())
  corr_trans_epi$label <- ifelse(corr_trans_epi$gid==1, corr_trans_epi$Symbol, "")
  ### Define relative size of points based on si
  corr_trans_epi$size <- 10+(-log10(corr_trans_epi$padj.y)/max(-log10(corr_trans_epi$padj.y)))
  

  labels <- ifelse(corr_trans_epi$s.annotation %in% roi & corr_trans_epi$padj.y <= padj.atac, paste(roi,collapse="|"),
                   ifelse(corr_trans_epi$padj.y <= padj.atac,
                          "Significant",
                          "NS")
                   )
  colors <- c(color.region.of.interest, color.border.sign.peaks, color.border.unsign.peaks)
  names(colors) <- c(paste(roi,collapse="|"), "Significant", "NS")
  corr_trans_epi$color <- ifelse(corr_trans_epi$s.annotation %in% roi & corr_trans_epi$padj.y <= padj.atac,
                                 color.region.of.interest,
                                 ifelse(corr_trans_epi$padj.y <= padj.atac,
                                        color.border.sign.peaks,
                                        color.border.unsign.peaks))
  
  corr_trans_epi$stroke <- ifelse(corr_trans_epi$s.annotation %in% roi & corr_trans_epi$padj.y <= padj.atac, stroke.high, stroke.std)
  
  
  q <- (quantile(corr_trans_epi$padj.y, legend.padj.point.size.percentiles))
  q <- q[order(q)]
  
  .convert <- function(corr_trans_epi, pv){
    return(10+(-log10(pv)/max(-log10(corr_trans_epi$padj.y))))
  }
  
  ### Return only data.
  if(return[1]=="data"){ return(corr_trans_epi)}
  ### Or either plot+data or data.
  else {
  ### Generate plot.
  p <- ggplot(corr_trans_epi, aes(x=ID, y=position, label=str_pad(label,12, side = "left"), fill=log2FoldChange.y, color=labels)) + 
    
         geom_point(shape=point.shape, aes(size=size,  stroke=stroke)) + ylim(ylim[1],ylim[2]) + 
    
         ### Define multiple scales of geometric points.
         scale_color_manual(values=colors) + 
         scale_size_continuous(corr_trans_epi$size, range = c(min.point.size, max.point.size), 
                               breaks=.convert(corr_trans_epi, q), 
                               labels=paste(str_pad(as.character(as.double(round(-log10(q),1))), width = 5, side="right"), 
                                            str_pad(paste("(",names(q),")",sep=""), width = 8, side = "right"),sep="")) + 
         scale_fill_gradient2(low=atac.regulation.color.low, mid="white", high=atac.regulation.color.high) +
         
         # Format legends
         guides(size=guide_legend(override.aes = aes(fill="black"), title.theme = element_text(face=legend.title.face, color=legend.title.color), title=legend.title.background.padj, order=2)) +
         guides(fill=guide_colorbar(title.theme = element_text(face=legend.title.face, color=legend.title.color), title=legend.title.background.fc, order = 1)) +
         guides(color=guide_legend(override.aes= aes(size=3),title.theme = element_text(face=legend.title.face, color=legend.title.color), title=legend.title.point.types, order=3))
  
  ### Theme adjusted
  p <- p + theme
  
  ### Remove x-axis ticks, text and line of both axis)
  p <- p +  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())+ theme(axis.line = element_blank()) +
  ### Set axis titles
  xlab(x.lab) + ylab(y.lab) +
  ### Gene name format
  geom_text(angle=90, hjust=move.gene.name.down, color=color.gene.name) +
  ### Add horizontal line at l2fc=0
  geom_hline(yintercept=0, linetype="dashed", color="black", alpha=0.5)
          
  ### Set and format plot title.
  p <- p + ggtitle(title) + theme(plot.title=element_text(face=title.face, color=title.color, hjust = title.hjust))
  
  ### Return plot and data.
  if(return[1]=="plot+data"){return(list(plot=p,data=corr_trans_epi))}
  ### Or only plot.
  else {return(p)}
  }
}

TP <- TowerPlot(deg_data = expression_data,
              dap_data = results, move.gene.name.down = 1.3,
              only.significant.peaks = F, top_n=25,
              point.shape=23, ylim=c(-6,8), min.point.size = .1, max.point.size = 12,
              color.gene.name = "black", roi=c("Promoter"),
              color.region.of.interest = "yellow", stack.inc=.3,
              color.border.sign.peaks = "black", stroke.high = .5, stroke.std = 0.5,
              color.border.unsign.peaks = "lightgrey", #title="Differential accessibility of Peaks in top 25 UP- and top 25 DOWN-regulated genes",
              theme=theme_cowplot(), return="plot")

TP
```



<h5> Correlation Plot </h5>

### All
```{r fig.width=10, fig.height=10, echo=F}
## Data
Table1 <- na.omit(results)
Table1 <- Table1 %>%
  group_by(condition, Symbol,  transcriptId) %>%
  summarise_at(vars(log2FoldChange, padj), mean)
Table1 <- as.data.frame(Table1)
Table2 <- expression_data
table1_name <- "ATAC-Seq"
table2_name <- "Transcriptome"

colnames(Table1) <- paste("first.",  colnames(Table1) , sep="") 
colnames(Table1)[2] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[2])
colnames(Table1)[3] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[3])
colnames(Table2) <- paste("second.",  colnames(Table2) , sep="") 
colnames(Table2)[1] <- gsub(pattern = "second.", replacement = "", x = colnames(Table2)[1])

### Configuration
ES1 <- 4
PV1 <- 5
ES2 <- 3
PV2 <- 7
SV <- c("Symbol")

## Logic
cut_lfc_1 <- 0
cut_padj_1 <- 1
cut_lfc_2 <- 0
cut_padj_2 <- 1
cut <- .56
include.background <- F

## UI

# Main correlation plot
pt.size=2
pt.shape=21
pt.border.color="black"
theme <- theme_classic()
both.up.color <- "red"
both.dn.color <- "steelblue"
up.dn.color <- "yellow"
dn.up.color <- "lightblue"
nol.sign.color <- "lightgrey"
xlab <- paste("Log2FoldChange (",table1_name,")",sep="")
ylab <- paste("Log2FoldChange (",table2_name,")",sep="")

# Marginal distribution
bins = 100
hist.border.color <- "black"
hist.color <- "white"
density.lwd <- 1
density.lt <- "solid"
density.lc <- "steelblue"

# Title
title.color = "black"
title.face = "bold"
title.hjust <- 0.4
title.font.size <- 20
title.text <- "Correlation plot - ATAC-Seq vs. RNA-Seq"
free.label.color <- "black"
free.label.fill <- "white"
label.pad.width <- 5


Table1 <- Table1[abs(Table1[,ES1]) >= cut_lfc_1 & Table1[,PV1] <= cut_padj_1,]
Table2 <- Table2[abs(Table2[,ES2]) >= cut_lfc_2 & Table2[,PV2] <= cut_padj_2,]
Table1 <- Table1[, c(SV, colnames(Table1)[ES1], colnames(Table1)[PV1])]
Table2 <- Table2[, c(SV, colnames(Table2)[ES2], colnames(Table2)[PV2])]

if(include.background){
  Background.Table1 <- DEG.ext[[2]]
  Background.Table2 <- DEG.ox.ext[[2]]
  colnames(Background.Table1)[1:8] <- paste("first.bg.",  colnames(Background.Table1)[1:8] , sep="") 
  colnames(Background.Table1)[1] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[1])
  colnames(Background.Table1)[8] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[8])
  colnames(Background.Table2)[1:8] <- paste("second.bg.",  colnames(Background.Table2)[1:8] , sep="") 
  colnames(Background.Table2)[1] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[1])
  colnames(Background.Table2)[8] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[8])
  BES1 <- 3
  BES2 <- 3
  BPV1 <- 7
  BPV2 <- 7
  Background.Table1 <- Background.Table1[abs(Background.Table1[, BES1]) >= cut & Background.Table1[, BPV1] <= cut_padj_1, ]
  Background.Table2 <- Background.Table2[abs(Background.Table2[, BES2]) >= cut & Background.Table2[, BPV2] <= cut_padj_2, ]
  Background.Table1 <- Background.Table1[, c(SV, colnames(Background.Table1)[BES1], colnames(Background.Table1)[BPV1])]
  Background.Table2 <- Background.Table2[, c(SV, colnames(Background.Table2)[BES2], colnames(Background.Table2)[BPV2])]
}
ES1 <- 3
PV1 <- 4
ES2 <- 3
PV2 <- 4
combined <- merge(Table1, Table2, by=SV)

if(include.background){
  combined <- merge(combined, Background.Table1, by=SV)
  combined <- merge(combined, Background.Table2, by=SV)
}

#combined[combined$first.log2FoldChange  0 & combined$first.bg.log2FoldChange > 0 & (combined$second.log2FoldChange < 0 & combined$second.bg.log2FoldChange > 0), ]



ES1 <- 2
PV1 <- 3
ES2 <- 4
PV2 <- 5

combined$up.up <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) >= cut, T, F)
combined$dn.dn <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) <= -cut, T, F)
combined$up.dn <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) <= -cut, T, F)
combined$dn.up <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) >= cut, T, F)

combined$up.ns <- ifelse((combined[,ES1]) >= cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.up <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) >= cut, T, F)
combined$dn.ns <- ifelse((combined[,ES1]) <= -cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.dn <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) <= -cut, T, F)

nr.up.up <- nrow(combined[combined$up.up,])
nr.dn.dn <- nrow(combined[combined$dn.dn,])
nr.up.dn <- nrow(combined[combined$up.dn,])
nr.dn.up <- nrow(combined[combined$dn.up,])
nr.up.ns <- nrow(combined[combined$up.ns,])
nr.ns.up <- nrow(combined[combined$ns.up,])
nr.dn.ns <- nrow(combined[combined$dn.ns,])
nr.ns.dn <- nrow(combined[combined$ns.dn,])
nr.else <- nrow(combined)-(nr.up.up+nr.dn.dn+nr.up.dn+nr.dn.up)-(nr.up.ns+nr.ns.up+nr.dn.ns+nr.ns.dn)


df <- data.frame(Upregulated=c(nr.up.up, nr.dn.up, nr.ns.up), Downregulated=c(nr.up.dn, nr.dn.dn, nr.ns.dn), NS_L2FC=c(nr.up.ns, nr.dn.ns, nr.else))
colnames(df) <- paste(colnames(df), " (",table1_name,")",sep="")
rownames(df) <- c("Upregulated", "Downregulated", "NS_L2FC")
rownames(df) <- paste(rownames(df), " (",table2_name,")",sep="")
df <- gridExtra::tableGrob(df, theme=gridExtra::ttheme_minimal())

combined$Relation <- ifelse(combined$up.up, "Up.Up", 
                            ifelse(combined$dn.dn, "Dn.Dn", 
                                   ifelse(combined$up.dn, "Up.Dn", 
                                          ifelse(combined$dn.up, "Dn.Up", 
                                                 ifelse(combined$up.ns, "Up.Ns", 
                                                        ifelse(combined$ns.up, "Ns.Up", 
                                                               ifelse(combined$dn.ns, "Dn.Ns", 
                                                                      ifelse(combined$ns.dn, "Ns.Dn", "Ns.Ns"))))))))
combined$Relation <- as.factor(combined$Relation)
named.color.vector <- c(both.up.color, both.dn.color, up.dn.color, dn.up.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color)
names(named.color.vector) <- c("Up.Up", "Dn.Dn", "Up.Dn", "Dn.Up", "Up.Ns", "Ns.Up", "Dn.Ns", "Ns.Dn", "Ns.Ns")


linetype= "dashed"

merge(combined, proteomics_data, by="Symbol")

## Main plot
p <- ggplot(combined, aes(x=first.log2FoldChange, y=second.log2FoldChange, fill=Relation)) + 
            geom_point(size=pt.size, shape = pt.shape, color = pt.border.color) + theme_classic() + 
            scale_fill_manual(values = named.color.vector) + 
            geom_hline(yintercept = cut, linetype=linetype) + geom_hline(yintercept = -cut, linetype=linetype) + 
            geom_vline(xintercept = cut, linetype=linetype) + geom_vline(xintercept = -cut, linetype=linetype) +
            xlab(xlab) + ylab(ylab)



tp <- ggplot(combined, aes(x=first.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic() + theme(axis.title.x = element_blank())

rp <- ggplot(combined, aes(x=second.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic()  + theme(axis.title.y = element_blank()) + coord_flip()

cowplot::plot_grid(
ggplotify::as.ggplot(cowplot::plot_grid(
                        cowplot::plot_grid(tp, ggplot()+theme_void(), rel_widths = c(12,2)),
                        cowplot::plot_grid(  p + theme(legend.position = "none"), rp, rel_widths = c(12,2)),
                     rel_heights = c(2,8), ncol=1)) + 
                     ggtitle(title.text) + theme(plot.title = element_text(face=title.face, 
                                                 size = title.font.size, 
                                                 hjust=title.hjust, 
                                                 colour = title.color)), df, rel_heights = c(4,1), ncol=1)



```


```{r fig.width=10, fig.height=10, echo=F}
Table1 <- na.omit(results)
Table1 <- Table1 %>%
  group_by(condition, Symbol,  transcriptId) %>%
  summarise_at(vars(log2FoldChange, padj), mean)
Table1 <- as.data.frame(Table1)


Table2 <- expression_data
table1_name <- "ATAC-Seq"
table2_name <- "Transcriptome"

source("R/functions.R")

ca <- correlation_analysis.2(dea_1=Table1, dea_2 = Table2, bg.dea_1 = NULL, bg.dea_2 = NULL, l2fc_pos_1 = 4, padj_pos_1 = 5, l2fc_pos_2 = 3, padj_pos_2 = 7, gene_fields = c("Symbol"), fg_regulation_criteria_1 = "de", fg_regulation_criteria_2 = "de", mode = "one", include.background = F, dea_1_name = table1_name, dea_2_name = table2_name, l2fc_1 = 0.56, l2fc_2 = 0.56)

ca$strict$correlation.plot
```


### ATAC-Seq significant.
```{r fig.width=10, fig.height=10, echo=F}
## Data
Table1 <- na.omit(results)
Table1 <- Table1 %>%
  group_by(condition, Symbol,  transcriptId) %>%
  summarise_at(vars(log2FoldChange, padj), mean)
Table1 <- as.data.frame(Table1)
Table2 <- expression_data
table1_name <- "ATAC-Seq"
table2_name <- "Transcriptome"

colnames(Table1) <- paste("first.",  colnames(Table1) , sep="") 
colnames(Table1)[2] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[2])
colnames(Table1)[3] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[3])
colnames(Table2) <- paste("second.",  colnames(Table2) , sep="") 
colnames(Table2)[1] <- gsub(pattern = "second.", replacement = "", x = colnames(Table2)[1])

### Configuration
ES1 <- 4
PV1 <- 5
ES2 <- 3
PV2 <- 7
SV <- c("Symbol")

## Logic
cut_lfc_1 <- 0
cut_padj_1 <- 0.05
cut_lfc_2 <- 0
cut_padj_2 <- 1
cut <- .56
include.background <- F

## UI

# Main correlation plot
pt.size=2
pt.shape=21
pt.border.color="black"
theme <- theme_classic()
both.up.color <- "red"
both.dn.color <- "steelblue"
up.dn.color <- "yellow"
dn.up.color <- "lightblue"
nol.sign.color <- "lightgrey"
xlab <- paste("Log2FoldChange (",table1_name,")",sep="")
ylab <- paste("Log2FoldChange (",table2_name,")",sep="")

# Marginal distribution
bins = 100
hist.border.color <- "black"
hist.color <- "white"
density.lwd <- 1
density.lt <- "solid"
density.lc <- "steelblue"

# Title
title.color = "black"
title.face = "bold"
title.hjust <- 0.4
title.font.size <- 20
title.text <- "Correlation plot - ATAC-Seq vs. RNA-Seq"
free.label.color <- "black"
free.label.fill <- "white"
label.pad.width <- 5


Table1 <- Table1[abs(Table1[,ES1]) >= cut_lfc_1 & Table1[,PV1] <= cut_padj_1,]
Table2 <- Table2[abs(Table2[,ES2]) >= cut_lfc_2 & Table2[,PV2] <= cut_padj_2,]
Table1 <- Table1[, c(SV, colnames(Table1)[ES1], colnames(Table1)[PV1])]
Table2 <- Table2[, c(SV, colnames(Table2)[ES2], colnames(Table2)[PV2])]

if(include.background){
  Background.Table1 <- DEG.ext[[2]]
  Background.Table2 <- DEG.ox.ext[[2]]
  colnames(Background.Table1)[1:8] <- paste("first.bg.",  colnames(Background.Table1)[1:8] , sep="") 
  colnames(Background.Table1)[1] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[1])
  colnames(Background.Table1)[8] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[8])
  colnames(Background.Table2)[1:8] <- paste("second.bg.",  colnames(Background.Table2)[1:8] , sep="") 
  colnames(Background.Table2)[1] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[1])
  colnames(Background.Table2)[8] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[8])
  BES1 <- 3
  BES2 <- 3
  BPV1 <- 7
  BPV2 <- 7
  Background.Table1 <- Background.Table1[abs(Background.Table1[, BES1]) >= cut & Background.Table1[, BPV1] <= cut_padj_1, ]
  Background.Table2 <- Background.Table2[abs(Background.Table2[, BES2]) >= cut & Background.Table2[, BPV2] <= cut_padj_2, ]
  Background.Table1 <- Background.Table1[, c(SV, colnames(Background.Table1)[BES1], colnames(Background.Table1)[BPV1])]
  Background.Table2 <- Background.Table2[, c(SV, colnames(Background.Table2)[BES2], colnames(Background.Table2)[BPV2])]
}
ES1 <- 3
PV1 <- 4
ES2 <- 3
PV2 <- 4
combined <- merge(Table1, Table2, by=SV)

if(include.background){
  combined <- merge(combined, Background.Table1, by=SV)
  combined <- merge(combined, Background.Table2, by=SV)
}

#combined[combined$first.log2FoldChange  0 & combined$first.bg.log2FoldChange > 0 & (combined$second.log2FoldChange < 0 & combined$second.bg.log2FoldChange > 0), ]



ES1 <- 2
PV1 <- 3
ES2 <- 4
PV2 <- 5

combined$up.up <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) >= cut, T, F)
combined$dn.dn <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) <= -cut, T, F)
combined$up.dn <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) <= -cut, T, F)
combined$dn.up <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) >= cut, T, F)

combined$up.ns <- ifelse((combined[,ES1]) >= cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.up <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) >= cut, T, F)
combined$dn.ns <- ifelse((combined[,ES1]) <= -cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.dn <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) <= -cut, T, F)

nr.up.up <- nrow(combined[combined$up.up,])
nr.dn.dn <- nrow(combined[combined$dn.dn,])
nr.up.dn <- nrow(combined[combined$up.dn,])
nr.dn.up <- nrow(combined[combined$dn.up,])
nr.up.ns <- nrow(combined[combined$up.ns,])
nr.ns.up <- nrow(combined[combined$ns.up,])
nr.dn.ns <- nrow(combined[combined$dn.ns,])
nr.ns.dn <- nrow(combined[combined$ns.dn,])
nr.else <- nrow(combined)-(nr.up.up+nr.dn.dn+nr.up.dn+nr.dn.up)-(nr.up.ns+nr.ns.up+nr.dn.ns+nr.ns.dn)


df <- data.frame(Upregulated=c(nr.up.up, nr.dn.up, nr.ns.up), Downregulated=c(nr.up.dn, nr.dn.dn, nr.ns.dn), NS_L2FC=c(nr.up.ns, nr.dn.ns, nr.else))
colnames(df) <- paste(colnames(df), " (",table1_name,")",sep="")
rownames(df) <- c("Upregulated", "Downregulated", "NS_L2FC")
rownames(df) <- paste(rownames(df), " (",table2_name,")",sep="")
df <- gridExtra::tableGrob(df, theme=gridExtra::ttheme_minimal())

combined$Relation <- ifelse(combined$up.up, "Up.Up", 
                            ifelse(combined$dn.dn, "Dn.Dn", 
                                   ifelse(combined$up.dn, "Up.Dn", 
                                          ifelse(combined$dn.up, "Dn.Up", 
                                                 ifelse(combined$up.ns, "Up.Ns", 
                                                        ifelse(combined$ns.up, "Ns.Up", 
                                                               ifelse(combined$dn.ns, "Dn.Ns", 
                                                                      ifelse(combined$ns.dn, "Ns.Dn", "Ns.Ns"))))))))
combined$Relation <- as.factor(combined$Relation)
named.color.vector <- c(both.up.color, both.dn.color, up.dn.color, dn.up.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color)
names(named.color.vector) <- c("Up.Up", "Dn.Dn", "Up.Dn", "Dn.Up", "Up.Ns", "Ns.Up", "Dn.Ns", "Ns.Dn", "Ns.Ns")


linetype= "dashed"

## Main plot
p <- ggplot(combined, aes(x=first.log2FoldChange, y=second.log2FoldChange, fill=Relation)) + 
            geom_point(size=pt.size, shape = pt.shape, color = pt.border.color) + theme_classic() + 
            scale_fill_manual(values = named.color.vector) + 
            geom_hline(yintercept = cut, linetype=linetype) + geom_hline(yintercept = -cut, linetype=linetype) + 
            geom_vline(xintercept = cut, linetype=linetype) + geom_vline(xintercept = -cut, linetype=linetype) +
            xlab(xlab) + ylab(ylab)



tp <- ggplot(combined, aes(x=first.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic() + theme(axis.title.x = element_blank())

rp <- ggplot(combined, aes(x=second.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic()  + theme(axis.title.y = element_blank()) + coord_flip()

cowplot::plot_grid(
ggplotify::as.ggplot(cowplot::plot_grid(
                        cowplot::plot_grid(tp, ggplot()+theme_void(), rel_widths = c(12,2)),
                        cowplot::plot_grid(  p + theme(legend.position = "none"), rp, rel_widths = c(12,2)),
                     rel_heights = c(2,8), ncol=1)) + 
                     ggtitle(title.text) + theme(plot.title = element_text(face=title.face, 
                                                 size = title.font.size, 
                                                 hjust=title.hjust, 
                                                 colour = title.color)), df, rel_heights = c(4,1), ncol=1)



```

### Transcriptomics significant
```{r fig.width=10, fig.height=10, echo=F}
## Data
Table1 <- na.omit(results)
Table1 <- Table1 %>%
  group_by(condition, Symbol,  transcriptId) %>%
  summarise_at(vars(log2FoldChange, padj), mean)
Table1 <- as.data.frame(Table1)
Table2 <- expression_data
table1_name <- "ATAC-Seq"
table2_name <- "Transcriptome"

colnames(Table1) <- paste("first.",  colnames(Table1) , sep="") 
colnames(Table1)[2] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[2])
colnames(Table1)[3] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[3])
colnames(Table2) <- paste("second.",  colnames(Table2) , sep="") 
colnames(Table2)[1] <- gsub(pattern = "second.", replacement = "", x = colnames(Table2)[1])

### Configuration
ES1 <- 4
PV1 <- 5
ES2 <- 3
PV2 <- 7
SV <- c("Symbol")

## Logic
cut_lfc_1 <- 0
cut_padj_1 <- 1
cut_lfc_2 <- 0
cut_padj_2 <- 0.05
cut <- .56
include.background <- F

## UI

# Main correlation plot
pt.size=2
pt.shape=21
pt.border.color="black"
theme <- theme_classic()
both.up.color <- "red"
both.dn.color <- "steelblue"
up.dn.color <- "yellow"
dn.up.color <- "lightblue"
nol.sign.color <- "lightgrey"
xlab <- paste("Log2FoldChange (",table1_name,")",sep="")
ylab <- paste("Log2FoldChange (",table2_name,")",sep="")

# Marginal distribution
bins = 100
hist.border.color <- "black"
hist.color <- "white"
density.lwd <- 1
density.lt <- "solid"
density.lc <- "steelblue"

# Title
title.color = "black"
title.face = "bold"
title.hjust <- 0.4
title.font.size <- 20
title.text <- "Correlation plot - ATAC-Seq vs. RNA-Seq"
free.label.color <- "black"
free.label.fill <- "white"
label.pad.width <- 5


Table1 <- Table1[abs(Table1[,ES1]) >= cut_lfc_1 & Table1[,PV1] <= cut_padj_1,]
Table2 <- Table2[abs(Table2[,ES2]) >= cut_lfc_2 & Table2[,PV2] <= cut_padj_2,]
Table1 <- Table1[, c(SV, colnames(Table1)[ES1], colnames(Table1)[PV1])]
Table2 <- Table2[, c(SV, colnames(Table2)[ES2], colnames(Table2)[PV2])]

if(include.background){
  Background.Table1 <- DEG.ext[[2]]
  Background.Table2 <- DEG.ox.ext[[2]]
  colnames(Background.Table1)[1:8] <- paste("first.bg.",  colnames(Background.Table1)[1:8] , sep="") 
  colnames(Background.Table1)[1] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[1])
  colnames(Background.Table1)[8] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[8])
  colnames(Background.Table2)[1:8] <- paste("second.bg.",  colnames(Background.Table2)[1:8] , sep="") 
  colnames(Background.Table2)[1] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[1])
  colnames(Background.Table2)[8] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[8])
  BES1 <- 3
  BES2 <- 3
  BPV1 <- 7
  BPV2 <- 7
  Background.Table1 <- Background.Table1[abs(Background.Table1[, BES1]) >= cut & Background.Table1[, BPV1] <= cut_padj_1, ]
  Background.Table2 <- Background.Table2[abs(Background.Table2[, BES2]) >= cut & Background.Table2[, BPV2] <= cut_padj_2, ]
  Background.Table1 <- Background.Table1[, c(SV, colnames(Background.Table1)[BES1], colnames(Background.Table1)[BPV1])]
  Background.Table2 <- Background.Table2[, c(SV, colnames(Background.Table2)[BES2], colnames(Background.Table2)[BPV2])]
}
ES1 <- 3
PV1 <- 4
ES2 <- 3
PV2 <- 4
combined <- merge(Table1, Table2, by=SV)

if(include.background){
  combined <- merge(combined, Background.Table1, by=SV)
  combined <- merge(combined, Background.Table2, by=SV)
}

#combined[combined$first.log2FoldChange  0 & combined$first.bg.log2FoldChange > 0 & (combined$second.log2FoldChange < 0 & combined$second.bg.log2FoldChange > 0), ]



ES1 <- 2
PV1 <- 3
ES2 <- 4
PV2 <- 5

combined$up.up <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) >= cut, T, F)
combined$dn.dn <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) <= -cut, T, F)
combined$up.dn <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) <= -cut, T, F)
combined$dn.up <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) >= cut, T, F)

combined$up.ns <- ifelse((combined[,ES1]) >= cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.up <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) >= cut, T, F)
combined$dn.ns <- ifelse((combined[,ES1]) <= -cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.dn <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) <= -cut, T, F)

nr.up.up <- nrow(combined[combined$up.up,])
nr.dn.dn <- nrow(combined[combined$dn.dn,])
nr.up.dn <- nrow(combined[combined$up.dn,])
nr.dn.up <- nrow(combined[combined$dn.up,])
nr.up.ns <- nrow(combined[combined$up.ns,])
nr.ns.up <- nrow(combined[combined$ns.up,])
nr.dn.ns <- nrow(combined[combined$dn.ns,])
nr.ns.dn <- nrow(combined[combined$ns.dn,])
nr.else <- nrow(combined)-(nr.up.up+nr.dn.dn+nr.up.dn+nr.dn.up)-(nr.up.ns+nr.ns.up+nr.dn.ns+nr.ns.dn)


df <- data.frame(Upregulated=c(nr.up.up, nr.dn.up, nr.ns.up), Downregulated=c(nr.up.dn, nr.dn.dn, nr.ns.dn), NS_L2FC=c(nr.up.ns, nr.dn.ns, nr.else))
colnames(df) <- paste(colnames(df), " (",table1_name,")",sep="")
rownames(df) <- c("Upregulated", "Downregulated", "NS_L2FC")
rownames(df) <- paste(rownames(df), " (",table2_name,")",sep="")
df <- gridExtra::tableGrob(df, theme=gridExtra::ttheme_minimal())

combined$Relation <- ifelse(combined$up.up, "Up.Up", 
                            ifelse(combined$dn.dn, "Dn.Dn", 
                                   ifelse(combined$up.dn, "Up.Dn", 
                                          ifelse(combined$dn.up, "Dn.Up", 
                                                 ifelse(combined$up.ns, "Up.Ns", 
                                                        ifelse(combined$ns.up, "Ns.Up", 
                                                               ifelse(combined$dn.ns, "Dn.Ns", 
                                                                      ifelse(combined$ns.dn, "Ns.Dn", "Ns.Ns"))))))))
combined$Relation <- as.factor(combined$Relation)
named.color.vector <- c(both.up.color, both.dn.color, up.dn.color, dn.up.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color)
names(named.color.vector) <- c("Up.Up", "Dn.Dn", "Up.Dn", "Dn.Up", "Up.Ns", "Ns.Up", "Dn.Ns", "Ns.Dn", "Ns.Ns")


linetype= "dashed"

## Main plot
p <- ggplot(combined, aes(x=first.log2FoldChange, y=second.log2FoldChange, fill=Relation)) + 
            geom_point(size=pt.size, shape = pt.shape, color = pt.border.color) + theme_classic() + 
            scale_fill_manual(values = named.color.vector) + 
            geom_hline(yintercept = cut, linetype=linetype) + geom_hline(yintercept = -cut, linetype=linetype) + 
            geom_vline(xintercept = cut, linetype=linetype) + geom_vline(xintercept = -cut, linetype=linetype) +
            xlab(xlab) + ylab(ylab)



tp <- ggplot(combined, aes(x=first.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic() + theme(axis.title.x = element_blank())

rp <- ggplot(combined, aes(x=second.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic()  + theme(axis.title.y = element_blank()) + coord_flip()

cowplot::plot_grid(
ggplotify::as.ggplot(cowplot::plot_grid(
                        cowplot::plot_grid(tp, ggplot()+theme_void(), rel_widths = c(12,2)),
                        cowplot::plot_grid(  p + theme(legend.position = "none"), rp, rel_widths = c(12,2)),
                     rel_heights = c(2,8), ncol=1)) + 
                     ggtitle(title.text) + theme(plot.title = element_text(face=title.face, 
                                                 size = title.font.size, 
                                                 hjust=title.hjust, 
                                                 colour = title.color)), df, rel_heights = c(4,1), ncol=1)



```

### Both significant
```{r fig.width=10, fig.height=10, echo=F, warning=F}
## Data
Table1 <- na.omit(results)
Table1 <- Table1 %>%
  group_by(condition, Symbol,  transcriptId) %>%
  summarise_at(vars(log2FoldChange, padj), mean)
Table1 <- as.data.frame(Table1)
Table2 <- expression_data
table1_name <- "ATAC-Seq"
table2_name <- "Transcriptome"

colnames(Table1) <- paste("first.",  colnames(Table1) , sep="") 
colnames(Table1)[2] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[2])
colnames(Table1)[3] <- gsub(pattern = "first.", replacement = "", x = colnames(Table1)[3])
colnames(Table2) <- paste("second.",  colnames(Table2) , sep="") 
colnames(Table2)[1] <- gsub(pattern = "second.", replacement = "", x = colnames(Table2)[1])

### Configuration
ES1 <- 4
PV1 <- 5
ES2 <- 3
PV2 <- 7
SV <- c("Symbol")

## Logic
cut_lfc_1 <- 0
cut_padj_1 <- 0.05
cut_lfc_2 <- 0
cut_padj_2 <- 0.05
cut <- .56
include.background <- F

## UI

# Main correlation plot
pt.size=2
pt.shape=21
pt.border.color="black"
theme <- theme_classic()
both.up.color <- "red"
both.dn.color <- "steelblue"
up.dn.color <- "yellow"
dn.up.color <- "lightblue"
nol.sign.color <- "lightgrey"
xlab <- paste("Log2FoldChange (",table1_name,")",sep="")
ylab <- paste("Log2FoldChange (",table2_name,")",sep="")

# Marginal distribution
bins = 100
hist.border.color <- "black"
hist.color <- "white"
density.lwd <- 1
density.lt <- "solid"
density.lc <- "steelblue"

# Title
title.color = "black"
title.face = "bold"
title.hjust <- 0.4
title.font.size <- 20
title.text <- "Correlation plot - ATAC-Seq vs. RNA-Seq"
free.label.color <- "black"
free.label.fill <- "white"
label.pad.width <- 5


Table1 <- Table1[abs(Table1[,ES1]) >= cut_lfc_1 & Table1[,PV1] <= cut_padj_1,]
Table2 <- Table2[abs(Table2[,ES2]) >= cut_lfc_2 & Table2[,PV2] <= cut_padj_2,]
Table1 <- Table1[, c(SV, colnames(Table1)[ES1], colnames(Table1)[PV1])]
Table2 <- Table2[, c(SV, colnames(Table2)[ES2], colnames(Table2)[PV2])]

if(include.background){
  Background.Table1 <- DEG.ext[[2]]
  Background.Table2 <- DEG.ox.ext[[2]]
  colnames(Background.Table1)[1:8] <- paste("first.bg.",  colnames(Background.Table1)[1:8] , sep="") 
  colnames(Background.Table1)[1] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[1])
  colnames(Background.Table1)[8] <- gsub(pattern = "first.bg.", replacement = "", x = colnames(Background.Table1)[8])
  colnames(Background.Table2)[1:8] <- paste("second.bg.",  colnames(Background.Table2)[1:8] , sep="") 
  colnames(Background.Table2)[1] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[1])
  colnames(Background.Table2)[8] <- gsub(pattern = "second.bg.", replacement = "", x = colnames(Background.Table2)[8])
  BES1 <- 3
  BES2 <- 3
  BPV1 <- 7
  BPV2 <- 7
  Background.Table1 <- Background.Table1[abs(Background.Table1[, BES1]) >= cut & Background.Table1[, BPV1] <= cut_padj_1, ]
  Background.Table2 <- Background.Table2[abs(Background.Table2[, BES2]) >= cut & Background.Table2[, BPV2] <= cut_padj_2, ]
  Background.Table1 <- Background.Table1[, c(SV, colnames(Background.Table1)[BES1], colnames(Background.Table1)[BPV1])]
  Background.Table2 <- Background.Table2[, c(SV, colnames(Background.Table2)[BES2], colnames(Background.Table2)[BPV2])]
}
ES1 <- 3
PV1 <- 4
ES2 <- 3
PV2 <- 4
combined <- merge(Table1, Table2, by=SV)

if(include.background){
  combined <- merge(combined, Background.Table1, by=SV)
  combined <- merge(combined, Background.Table2, by=SV)
}

#combined[combined$first.log2FoldChange  0 & combined$first.bg.log2FoldChange > 0 & (combined$second.log2FoldChange < 0 & combined$second.bg.log2FoldChange > 0), ]



ES1 <- 2
PV1 <- 3
ES2 <- 4
PV2 <- 5

combined$up.up <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) >= cut, T, F)
combined$dn.dn <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) <= -cut, T, F)
combined$up.dn <- ifelse((combined[,ES1]) >= cut & (combined[,ES2]) <= -cut, T, F)
combined$dn.up <- ifelse((combined[,ES1]) <= -cut & (combined[,ES2]) >= cut, T, F)

combined$up.ns <- ifelse((combined[,ES1]) >= cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.up <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) >= cut, T, F)
combined$dn.ns <- ifelse((combined[,ES1]) <= -cut & ((combined[,ES2]) >= -cut &(combined[,ES2]) <= cut), T, F)
combined$ns.dn <- ifelse(((combined[,ES1]) >= -cut &(combined[,ES1]) <= cut) & (combined[,ES2]) <= -cut, T, F)

nr.up.up <- nrow(combined[combined$up.up,])
nr.dn.dn <- nrow(combined[combined$dn.dn,])
nr.up.dn <- nrow(combined[combined$up.dn,])
nr.dn.up <- nrow(combined[combined$dn.up,])
nr.up.ns <- nrow(combined[combined$up.ns,])
nr.ns.up <- nrow(combined[combined$ns.up,])
nr.dn.ns <- nrow(combined[combined$dn.ns,])
nr.ns.dn <- nrow(combined[combined$ns.dn,])
nr.else <- nrow(combined)-(nr.up.up+nr.dn.dn+nr.up.dn+nr.dn.up)-(nr.up.ns+nr.ns.up+nr.dn.ns+nr.ns.dn)


df <- data.frame(Upregulated=c(nr.up.up, nr.dn.up, nr.ns.up), Downregulated=c(nr.up.dn, nr.dn.dn, nr.ns.dn), NS_L2FC=c(nr.up.ns, nr.dn.ns, nr.else))
colnames(df) <- paste(colnames(df), " (",table1_name,")",sep="")
rownames(df) <- c("Upregulated", "Downregulated", "NS_L2FC")
rownames(df) <- paste(rownames(df), " (",table2_name,")",sep="")
df <- gridExtra::tableGrob(df, theme=gridExtra::ttheme_minimal())

combined$Relation <- ifelse(combined$up.up, "Up.Up", 
                            ifelse(combined$dn.dn, "Dn.Dn", 
                                   ifelse(combined$up.dn, "Up.Dn", 
                                          ifelse(combined$dn.up, "Dn.Up", 
                                                 ifelse(combined$up.ns, "Up.Ns", 
                                                        ifelse(combined$ns.up, "Ns.Up", 
                                                               ifelse(combined$dn.ns, "Dn.Ns", 
                                                                      ifelse(combined$ns.dn, "Ns.Dn", "Ns.Ns"))))))))
combined$Relation <- as.factor(combined$Relation)
named.color.vector <- c(both.up.color, both.dn.color, up.dn.color, dn.up.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color, nol.sign.color)
names(named.color.vector) <- c("Up.Up", "Dn.Dn", "Up.Dn", "Dn.Up", "Up.Ns", "Ns.Up", "Dn.Ns", "Ns.Dn", "Ns.Ns")


linetype= "dashed"

## Main plot
p <- ggplot(combined, aes(x=first.log2FoldChange, y=second.log2FoldChange, fill=Relation, label=Symbol)) + 
            geom_point(size=pt.size, shape = pt.shape, color = pt.border.color) + geom_text_repel(max.overlaps = 20, text.size = 3, pt.border.color = "lightgrey", max.overlap = 200000, force = 0.25, force_pull = 3, min.segment.length = 0.1, segment.linetype = "dotted", segment.alpha = 1, segment.ncp = 3, segment.color = "white") + theme_classic() + 
            scale_fill_manual(values = named.color.vector) + 
            geom_hline(yintercept = cut, linetype=linetype) + geom_hline(yintercept = -cut, linetype=linetype) + 
            geom_vline(xintercept = cut, linetype=linetype) + geom_vline(xintercept = -cut, linetype=linetype) +
            xlab(xlab) + ylab(ylab)



tp <- ggplot(combined, aes(x=first.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic() + theme(axis.title.x = element_blank())

rp <- ggplot(combined, aes(x=second.log2FoldChange)) + 
        geom_histogram(aes(y = ..density..), bins = bins, colour = hist.border.color, fill = hist.color) + 
        geom_density(lwd = density.lwd, linetype = density.lt,  colour = density.lc)+ 
        theme_classic()  + theme(axis.title.y = element_blank()) + coord_flip()

cowplot::plot_grid(
ggplotify::as.ggplot(cowplot::plot_grid(
                        cowplot::plot_grid(tp, ggplot()+theme_void(), rel_widths = c(12,2)),
                        cowplot::plot_grid(  p + theme(legend.position = "none"), rp, rel_widths = c(12,2)),
                     rel_heights = c(2,8), ncol=1)) + 
                     ggtitle(title.text) + theme(plot.title = element_text(face=title.face, 
                                                 size = title.font.size, 
                                                 hjust=title.hjust, 
                                                 colour = title.color)), df, rel_heights = c(4,1), ncol=1)



```



## {.tabset .tabset-fade}

<h3>5. Motif matching and footprinting </h3>


<h4> 5.1 Volcano plots of top TF targets</h4>

### STAT1
```{r  fig.height=6, fig.width=10, warning=F, echo=F}

STAT1.JSON <- fromJSON(file="data/datasets/STAT1_Targets.txt")
targets <- vector("list", length(STAT1.JSON$associations))
for(g in 1:length(STAT1.JSON$associations)){
  targets[[g]] <- STAT1.JSON$associations[[g]]$gene$symbol
}
targets <- unlist(targets)

selected.genes <- results[results$Symbol %in% targets,]
selected.genes <- selected.genes[abs(selected.genes$log2FoldChange) >= 1 & (selected.genes$padj) <= 0.01 ,]
selected.genes <- selected.genes[order(selected.genes$padj),]
selected.genes <- na.omit(selected.genes)
selected.genes[selected.genes$condition=="Lyn.WT",]$log2FoldChange <- -selected.genes[selected.genes$condition=="Lyn.WT",]$log2FoldChange
selected.genes <- selected.genes %>% group_by(condition) %>% top_n(10, log2FoldChange)
selected.genes <- selected.genes$Symbol

plot_Volcano(DEG.table = results, selected.genes = selected.genes,   colors.selected.genes = c("black"), xlab="log2(FC) - Regulation of accessibility",pt.size = 2,repel.text.size = 3, point.colors = point.colors, pt.border.color = "lightgrey", repel.max.overlap = 200000, repel.force = 0.25, repel.force_pull = 3, repel.min.segment.length = 0.1, repel.segment.linetype = "dotted", repel.segment.alpha = 1, repel.segment.ncp = 3, repel.segment.color = "black", ylim=c(0, 20), xlim=c(-5,5), title="Interval/Peak Level", title.color = "darkred")
```

### AP-1
```{r  fig.height=6, fig.width=10, warning=F, echo=F}

AP1.JSON <- fromJSON(file="data/datasets/AP1_Targets.txt")
targets <- vector("list", length(AP1.JSON$associations))
for(g in 1:length(AP1.JSON$associations)){
  targets[[g]] <- AP1.JSON$associations[[g]]$gene$symbol
}
targets <- unlist(targets)

selected.genes <- results[results$Symbol %in% targets,]
selected.genes <- selected.genes[abs(selected.genes$log2FoldChange) >= 1 & (selected.genes$padj) <= 0.01 ,]
selected.genes <- selected.genes[order(selected.genes$padj),]
selected.genes <- na.omit(selected.genes)
selected.genes[selected.genes$condition=="Lyn.WT",]$log2FoldChange <- -selected.genes[selected.genes$condition=="Lyn.WT",]$log2FoldChange
selected.genes <- selected.genes %>% group_by(condition) %>% top_n(10, log2FoldChange)
selected.genes <- selected.genes$Symbol

plot_Volcano(DEG.table = results, selected.genes = selected.genes,   colors.selected.genes = c("black"), xlab="log2(FC) - Regulation of accessibility",pt.size = 2,repel.text.size = 3, point.colors = point.colors, pt.border.color = "lightgrey", repel.max.overlap = 200000, repel.force = 0.25, repel.force_pull = 3, repel.min.segment.length = 0.1, repel.segment.linetype = "dotted", repel.segment.alpha = 1, repel.segment.ncp = 3, repel.segment.color = "black", ylim=c(0, 20), xlim=c(-5,5), title="Interval/Peak Level", title.color = "darkred")
```

```{r echo=F}
lyn.bed <- read.table("/home/ali_taleb/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/motif.analysis/indir/input/Lyn-KO.mRp.clN_summits.bed", sep="\t")
colnames(lyn.bed) <- c("Seq", "Start", "Ende", "Peak", "Summit")
lyn.bed$Seq <- paste("chr",lyn.bed$Seq,sep="")

write.table(lyn.bed, file = "/home/ali_taleb/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/motif.analysis/indir/input/Lyn.bed", col.names = F, sep="\t", quote=F, row.names = F)


wt.bed <- read.table("/home/ali_taleb/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/motif.analysis/indir/input/Lyn-KO.mRp.clN_summits.bed", sep="\t")
colnames(wt.bed) <- c("Seq", "Start", "Ende", "Peak", "Summit")
wt.bed$Seq <- paste("chr",wt.bed$Seq,sep="")

write.table(wt.bed, file = "/home/ali_taleb/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/motif.analysis/indir/input/WT.bed", sep="\t", col.names = F, quote=F, row.names = F)
```



## {.tabset .tabset-fade}
<h4> 5.2 Motif-based analysis</h4>
```{r echo=F, warning=F}
motifanalysis_dir <- "~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/motif.analysis/motif.analysis"
roi <- paste(motifanalysis_dir, "/enrichment/roi/",sep="")
match <- paste(motifanalysis_dir, "/match/",sep="")

fulltest_statistics_file <- paste(roi, "/fulltest_statistics.txt",sep="")
bed_file <- paste(match, "/roi_mpbs.bed",sep="")
ft_statsitics_df <- fulltest_statistics_file %>% read.table(header = T, sep = "\t")
Table_title <- "Motif enrichment analysis"
table_name <- "Motif_enrichment_analysis"
DT::datatable(ft_statsitics_df, rownames = FALSE, filter = 'top', 
                                       caption = Table_title, extensions = 'Buttons',
                                       options = list(dom = 'Bfrtip', scrollX=T,                                                                                                                          
                                       buttons = list(list(extend='csv', title=table_name), 
                                                      list(extend='excel', filename=table_name),    
                                                      list(extend='pdf', filename=table_name)), 
                                       autoWidth = TRUE, pageLength = 20, 
                                       columnDefs = list(list(className = 'dt-left'))))


```




## {.tabset .tabset-fade}
<h4> 5.3 Footprinting</h4>
```{r echo=F}

footprinting_dir <- "~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/"
diff_footprints_dir <- paste(footprinting_dir, "/4_differential_footprints/LynKO_LynWT/", sep="")

activity_plot <- read.table(file = paste(diff_footprints_dir, "/differential_statistics.txt", sep=""), header = T, sep = "\t")

```

<h5>Summary</h5>

### Combined
```{r echo=F, fig.width=20, fig.height=7, warning=F}  
footprinting.plot.norm <- footprinting_plot(activity_df=activity_plot, repel="label", repel.text.color="black", repel.text.size = 3, selected.genes = c("STAT1"), repel.max.overlaps = 2000000, pt.sizes = c(2,2,2,2), title="Activity score distribution",  title.color = "#000080", title.face = "plain")[[1]]

footprinting.plot.volcano <- footprinting_plot(activity_df=activity_plot, activity = "Z_score", log10.second.variable=T, second.variable = "P_values", selected.genes = c("STAT1"), ylim=c(-5,5), pt.sizes = c(2,0.5,2,2), ylab = "KO <-- z-score of Activity Score --> WT", xlab= "-log10(adjusted p-value) [significance]" ,sign.hline = "both", sign.hcut = 1, sign.vcut = -log10(0.05), sign.vline = "one", coord_flip = T, title="Activity score vs. Adjustet p. value", title.color = "#000080", title.face = "plain")[[1]]

ggplotify::as.ggplot(cowplot::plot_grid(cowplot::plot_grid(footprinting.plot.norm, ggplot()+theme_void(), ncol=1, rel_heights = c(40,1)), footprinting.plot.volcano, ncol=2)) + ggtitle("Footprinting analysis - Summary") + theme(plot.title = element_text(hjust= 0.5, face = "bold", color="#800080", size=20))
```

### Activity Plot
```{r echo=F, fig.width=10, fig.height=7, warning=F}  
footprinting.plot.norm
```


### Volcano-Like Plot
```{r echo=F, fig.width=10, fig.height=7, warning=F}  
footprinting.plot.volcano
```

## {.tabset .tabset-fade}

<h5>LinePlots (AP-1 subunits)  </h5>

```{r echo=F, fig.width=18, fig.height=18}
lineplots_dir <- "~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/4_differential_footprints/LynKO_LynWT/Lineplots/"
lineplots_file <- list.files(path = lineplots_dir, pattern = "*txt", full.names = T)
lineplots_names <- list.files(path = lineplots_dir, pattern = "*txt", full.names = F)
roi_lp_data_file_path <- as.data.frame(lineplots_file) %>% filter(grepl("NFKB|STAT1|JUN|FOS|(^ATF)",lineplots_file)) %>% as.list() %>% unlist()
roi_lp_data_names <- as.data.frame(lineplots_names) %>% filter(grepl("NFKB|THBS1|STAT1|JUN|FOS|(^ATF)",lineplots_file)) %>% as.list() %>% unlist() %>% gsub(pattern = "\\.txt", replacement = "")
lineplots <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  df <- read.table(roi_lp_data_file_path[r], header = T, sep="\t")
  df$id <- -100:99
  df <- df %>% tidyr::gather(key="Condition", value="Coverage", 1:2)
  lineplots[[r]] <- ggplot(df, aes(x=id, y=Coverage, colour=Condition, linetype=Condition)) + 
                    geom_line() + theme_pubr() + 
                    ylab("Coverage") + xlab("Position around footprint") + 
                    ggtitle(roi_lp_data_names[r]) + theme(plot.title = element_text(hjust=0.5, face="bold")) + 
                    scale_color_manual(values=c("grey", "red")) + scale_linetype_manual(values=c("solid", "solid"))
}

lines <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  
  very.strongly.sign.motifs <- activity_plot[activity_plot$P_values<0.001,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  strongly.sign.motifs <- activity_plot[activity_plot$P_values<0.01,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  sign.motifs <- activity_plot[activity_plot$P_values<0.05,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  weakly.sign.motifs <- activity_plot[activity_plot$P_values<0.1,]$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  
  all.motifs <- activity_plot$Motif %>% 
            gsub(pattern = "\\(", replacement = "_")  %>% 
            gsub(pattern = "\\)", replacement = "")
  
  activity_plot$Motif.2 <- all.motifs
  
  standard.titles <- paste(roi_lp_data_names[r], " (",round(activity_plot[activity_plot$Motif.2 %in% c(roi_lp_data_names[r]),]$P_values, 2),")",sep="")
  
  sign.titles <- paste(standard.titles," *",sep="")
  strongly.sign.titles <- paste(standard.titles," **",sep="")
  very.strongly.sign.titles <- paste(standard.titles," ***",sep="")
  weak.sign.titles <- paste(standard.titles," +",sep="")
 
  if(F){ 
  writeLines( c(paste("### ",ifelse(roi_lp_data_names[r] %in% very.strongly.sign.motifs, 
                                    very.strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% strongly.sign.motifs, 
                                    strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% sign.motifs, 
                                           sign.titles,
                                           ifelse(roi_lp_data_names[r] %in% weakly.sign.motifs, 
                                                  weak.sign.titles,
                                                  standard.titles)))
                                    ),
                      sep=""), 
               "```{r echo=F}", 
               paste("lineplots[[",r,"]]",sep=""),
               "```",
               "")
  )
  }
 
}
```

### MA0099.3.FOS::JUN (0.88)
```{r echo=F}
lineplots[[1]]
```

### MA0105.4.NFKB1 (0.75)
```{r echo=F}
lineplots[[2]]
```

### MA0137.3.STAT1 (0.55)
```{r echo=F}
lineplots[[3]]
```

### MA0462.2.BATF::JUN (0.69)
```{r echo=F}
lineplots[[4]]
```

### MA0476.1.FOS (0.23)
```{r echo=F}
lineplots[[5]]
```

### MA0477.2.FOSL1 (0.62)
```{r echo=F}
lineplots[[6]]
```

### MA0478.1.FOSL2 (0.91)
```{r echo=F}
lineplots[[7]]
```

### MA0488.1.JUN (0.08) +
```{r echo=F}
lineplots[[8]]
```

### MA0489.1.JUN_var.2 (0.72)
```{r echo=F}
lineplots[[9]]
```

### MA0490.2.JUNB (0.02) *
```{r echo=F}
lineplots[[10]]
```

### MA0491.2.JUND (0.03) *
```{r echo=F}
lineplots[[11]]
```

### MA0492.1.JUND_var.2 (0.04) *
```{r echo=F}
lineplots[[12]]
```

### MA0517.1.STAT1::STAT2 (0.11)
```{r echo=F}
lineplots[[13]]
```

### MA0778.1.NFKB2 (0.42)
```{r echo=F}
lineplots[[14]]
```

### MA1126.1.FOS::JUN_var.2 (0.04) *
```{r echo=F}
lineplots[[15]]
```

### MA1127.1.FOSB::JUN (0.02) *
```{r echo=F}
lineplots[[16]]
```

### MA1128.1.FOSL1::JUN (0.09) +
```{r echo=F}
lineplots[[17]]
```

### MA1129.1.FOSL1::JUN_var.2 (0.01) **
```{r echo=F}
lineplots[[18]]
```

### MA1130.1.FOSL2::JUN (0.88)
```{r echo=F}
lineplots[[19]]
```

### MA1131.1.FOSL2::JUN_var.2 (0.08) +
```{r echo=F}
lineplots[[20]]
```

### MA1132.1.JUN::JUNB (0.73)
```{r echo=F}
lineplots[[21]]
```

### MA1133.1.JUN::JUNB_var.2 (0.06) +
```{r echo=F}
lineplots[[22]]
```

### MA1134.1.FOS::JUNB (0.84)
```{r echo=F}
lineplots[[23]]
```

### MA1135.1.FOSB::JUNB (0.63)
```{r echo=F}
lineplots[[24]]
```

### MA1136.1.FOSB::JUNB_var.2 (0.006) ***
```{r echo=F}
lineplots[[25]]
```

### MA1137.1.FOSL1::JUNB (0.29)
```{r echo=F}
lineplots[[26]]
```

### MA1138.1.FOSL2::JUNB (0.29)
```{r echo=F}
lineplots[[27]]
```

### MA1139.1.FOSL2::JUNB_var.2 (0.01) **
```{r echo=F}
lineplots[[28]]
```

### MA1140.2.JUNB_var.2 (0.49)
```{r echo=F}
lineplots[[29]]
```

### MA1141.1.FOS::JUND (0.38)
```{r echo=F}
lineplots[[30]]
```

### MA1142.1.FOSL1::JUND (0.01) **
```{r echo=F}
lineplots[[31]]
```

### MA1143.1.FOSL1::JUND_var.2 (0.01) **
```{r echo=F}
lineplots[[32]]
```

### MA1145.1.FOSL2::JUND_var.2 (0.02) *
```{r echo=F}
lineplots[[33]]
```
## {.tabset .tabset-fade}

<h5>LinePlots (differentially binding TF)  </h5>

```{r echo=F}
"MA0746.2.SP3|MA0888.1.EVX2|MA0862.1.GMEB2|MA0604.1.Atf1|MA0060.3.NFYA|MA0844.1.XBP1|MA1564.1.SP9|MA1475.1.CREB3L4(var.2)|MA0840.1.Creb5|MA1142.1.FOSL1::JUND|MA0834.1.ATF7|MA0837.1.CEBPE|MA1143.1.FOSL1::JUND(var.2)|MA0886.1.EMX2|MA1634.1.BATF|MA1125.1.ZNF384|MA0685.1.SP4|MA1683.1.FOXA3|MA1126.1.FOS::JUN(var.2)|MA0835.2.BATF3|MA1145.1.FOSL2::JUND(var.2)|MA0490.2.JUNB|MA0516.2.SP2|MA0638.1.CREB3|MA0492.1.JUND(var.2)|MA1136.1.FOSB::JUNB(var.2)|MA1560.1.SOHLH2|MA0491.2.JUND|MA0893.2.GSX2|MA0502.2.NFYB|MA1470.1.BACH2(var.2)|MA0912.2.HOXD3|MA0079.4.SP1|MA0050.2.IRF1|MA1632.1.ATF2|MA1127.1.FOSB::JUN|MA0740.1.KLF14|MA0464.2.BHLHE40|MA0609.2.CREM|MA1557.1.SMAD5|MA1129.1.FOSL1::JUN(var.2)|MA0755.1.CUX2|MA1139.1.FOSL2::JUNB(var.2)"


```


```{r echo=F, fig.width=18, fig.height=18}
lineplots_dir <- "~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/4_differential_footprints/LynKO_LynWT/Lineplots/"
diff_statistics <- read.table("~/Desktop/RProjects/ATAC_Reprogramming_of_Fibroblasts/results/footprinting/4_differential_footprints/LynKO_LynWT/differential_statistics.txt", sep = "\t", header = T)
diff_statistics <- diff_statistics %>% filter(P_values <= 0.05)
lineplots_file <- list.files(path = lineplots_dir, pattern = "*txt", full.names = T)
lineplots_names <- list.files(path = lineplots_dir, pattern = "*txt", full.names = F)



roi_lp_data_file_path <- as.data.frame(lineplots_file) %>% filter(grepl(paste("(",paste(diff_statistics$Motif,collapse = ")|("),")",sep=""),lineplots_file)) %>% as.list() %>% unlist()
roi_lp_data_names <- as.data.frame(lineplots_names) %>% filter(grepl(paste("(",paste(diff_statistics$Motif,collapse = ")|("),")",sep=""),lineplots_file)) %>% as.list() %>% unlist() %>% gsub(pattern = "\\.txt", replacement = "")
lineplots <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  df <- read.table(roi_lp_data_file_path[r], header = T, sep="\t")
  df$id <- -100:99
  df <- df %>% tidyr::gather(key="Condition", value="Coverage", 1:2)
  lineplots[[r]] <- ggplot(df, aes(x=id, y=Coverage, colour=Condition, linetype=Condition)) + 
                    geom_line() + theme_pubr() + 
                    ylab("Coverage") + xlab("Position around footprint") + 
                    ggtitle(roi_lp_data_names[r]) + theme(plot.title = element_text(hjust=0.5, face="bold")) + 
                    scale_color_manual(values=c("grey", "red")) + scale_linetype_manual(values=c("solid", "solid"))
}

lines <- vector("list", length(roi_lp_data_file_path))
for(r in 1:length(roi_lp_data_file_path)){
  
  very.strongly.sign.motifs <-  activity_plot[activity_plot$P_values<0.001,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
       strongly.sign.motifs <-  activity_plot[activity_plot$P_values<0.01,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
                sign.motifs <-  activity_plot[activity_plot$P_values<0.05,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
         weakly.sign.motifs <-  activity_plot[activity_plot$P_values<0.1,]$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
                 all.motifs <-  activity_plot$Motif %>% 
                                gsub(pattern = "\\(", replacement = "_")  %>% 
                                gsub(pattern = "\\)", replacement = "")
  
  activity_plot$Motif.2 <- all.motifs
  standard.titles <- paste(roi_lp_data_names[r]," ",
                           "(",
                           round(activity_plot[activity_plot$Motif.2 %in% c(roi_lp_data_names[r]),]$P_values, 2),
                           ")",
                           sep="")
  sign.titles <- paste(standard.titles," *",sep="")
  strongly.sign.titles <- paste(standard.titles," **",sep="")
  very.strongly.sign.titles <- paste(standard.titles," ***",sep="")
  weak.sign.titles <- paste(standard.titles," +",sep="")
  
  writeLines( c(paste("### ",ifelse(roi_lp_data_names[r] %in% very.strongly.sign.motifs, 
                                    very.strongly.sign.titles,
                                    ifelse(roi_lp_data_names[r] %in% strongly.sign.motifs,
                                           strongly.sign.titles,
                                           ifelse(roi_lp_data_names[r] %in% sign.motifs,
                                                  sign.titles,
                                                  ifelse(roi_lp_data_names[r] %in% weakly.sign.motifs,
                                                         weak.sign.titles,
                                                         standard.titles)))),sep=""), 
                     "```{r echo=F}", 
                     paste("lineplots[[",r,"]]",sep=""),
                     "```",
                     ""))
}
```




### MA0050.2.IRF1 (0) ***
```{r echo=F}
lineplots[[1]]
```

### MA0060.3.NFYA (0) **
```{r echo=F}
lineplots[[2]]
```

### MA0079.4.SP1 (0.01) *
```{r echo=F}
lineplots[[3]]
```

### MA0464.2.BHLHE40 (0.02) *
```{r echo=F}
lineplots[[4]]
```

### MA0490.2.JUNB (0.02) *
```{r echo=F}
lineplots[[5]]
```

### MA0491.2.JUND (0.03) *
```{r echo=F}
lineplots[[6]]
```

### MA0502.2.NFYB (0) ***
```{r echo=F}
lineplots[[7]]
```

### MA0516.2.SP2 (0.01) *
```{r echo=F}
lineplots[[8]]
```

### MA0604.1.Atf1 (0.02) *
```{r echo=F}
lineplots[[9]]
```

### MA0609.2.CREM (0.04) *
```{r echo=F}
lineplots[[10]]
```

### MA0638.1.CREB3 (0.03) *
```{r echo=F}
lineplots[[11]]
```

### MA0685.1.SP4 (0) **
```{r echo=F}
lineplots[[12]]
```

### MA0740.1.KLF14 (0) **
```{r echo=F}
lineplots[[13]]
```

### MA0746.2.SP3 (0.02) *
```{r echo=F}
lineplots[[14]]
```

### MA0755.1.CUX2 (0.01) **
```{r echo=F}
lineplots[[15]]
```

### MA0834.1.ATF7 (0.04) *
```{r echo=F}
lineplots[[16]]
```

### MA0835.2.BATF3 (0.02) *
```{r echo=F}
lineplots[[17]]
```

### MA0837.1.CEBPE (0.02) *
```{r echo=F}
lineplots[[18]]
```

### MA0840.1.Creb5 (0.04) *
```{r echo=F}
lineplots[[19]]
```

### MA0844.1.XBP1 (0.01) **
```{r echo=F}
lineplots[[20]]
```

### MA0862.1.GMEB2 (0.01) **
```{r echo=F}
lineplots[[21]]
```

### MA0886.1.EMX2 (0.03) *
```{r echo=F}
lineplots[[22]]
```

### MA0888.1.EVX2 (0) ***
```{r echo=F}
lineplots[[23]]
```

### MA0893.2.GSX2 (0) **
```{r echo=F}
lineplots[[24]]
```

### MA0912.2.HOXD3 (0.03) *
```{r echo=F}
lineplots[[25]]
```

### MA1125.1.ZNF384 (0) **
```{r echo=F}
lineplots[[26]]
```

### MA1127.1.FOSB::JUN (0.02) *
```{r echo=F}
lineplots[[27]]
```

### MA1142.1.FOSL1::JUND (0.01) **
```{r echo=F}
lineplots[[28]]
```

### MA1557.1.SMAD5 (0.04) *
```{r echo=F}
lineplots[[29]]
```

### MA1560.1.SOHLH2 (0) ***
```{r echo=F}
lineplots[[30]]
```

### MA1564.1.SP9 (0.04) *
```{r echo=F}
lineplots[[31]]
```

### MA1632.1.ATF2 (0.03) *
```{r echo=F}
lineplots[[32]]
```

### MA1634.1.BATF (0) ***
```{r echo=F}
lineplots[[33]]
```

### MA1683.1.FOXA3 (0) ***
```{r echo=F}
lineplots[[34]]
```

### Author and session Information
<h4>Analyzed by</h4>
Dr. Ali T. Abdallah, Dipl. Inf.</br>
Bioinformatics Facility (Coordinator)</br>
University Hospital of Cologne - CECAD Research Center</br>
Joseph-Stelzmann-Str. 26, 52074 Cologne, Germany</br>
Phone +49 221 478 84017</br>

<h4>Detailed Session Info</h4>
```{r echo=FALSE}
sessionInfo()
```